---
title: "depMapDataCleaning"
author: "Theo Killian"
date: "March 11, 2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# [depmap](https://github.com/tfkillian/depmap/) data cleaning

This is a vignette displaying the how the 
[depmap](https://github.com/tfkillian/depmap/) was obtained and cleaned before 
being included in the final [depmap](https://github.com/tfkillian/depmap/)
package.  

## Introduction {#sec:intro}

The [depmap](https://github.com/tfkillian/depmap/) package aims to provide a 
reproducible research framework to cancer dependency data described by 
[Tsherniak, Aviad, et al. "Defining a cancer dependency map." Cell 170.3 (2017): 564-576.](https://www.ncbi.nlm.nih.gov/pubmed/28753430). The 
[depmap](https://github.com/tfkillian/depmap/) package should allow a researcher 
to easily mine dependency data taken from this cancer genomic study, explore the 
data and its statistical properties and visually display them.

```{r, eval=TRUE, echo=F}
# this is a function to ensure all the necessary packages are loaded to run this
# vignette
packages <- c("RCurl", "dplyr")
if ( length(missing_pkgs <- setdiff(packages, 
                                    rownames(installed.packages()))) > 0) {
  message("Installing missing package(s): ", 
          paste(missing_pkgs, collapse = ", "))
  install.packages(missing_pkgs)
}
```

## Downloading [Broad Institute](https://depmap.org/portal/download/) data 

If desired, the [depmap](https://github.com/tfkillian/depmap/) datasets can be 
downloaded in *.csv* format directly from the 
[Broad Institute](https://depmap.org/portal/download/) website. However, be 
aware that these files are 1.5GB in total (non-compressed) and take a moderate 
amount of time to download!

Users who wish to download the raw DRIVE data from which the DepMap data was 
been derived can be downloaded from the following here:
https://data.mendeley.com/datasets/y3ds55n88r/4

Note: the raw DRIVE data has not been subjected to dependency correction using
CERES or DEMETER2 algorithms. The code for CERES, DEMETER and other DepMap
algorithms can be found at: https://github.com/cancerdatasci

```{r, eval=FALSE, echo=T}
library("RCurl", warn.conflicts = FALSE)

# r read csv from url allows direct download csv file from website

# download depmap_19Q1_mutation_calls.csv
depmap_19Q1_mutation_calls <- read.csv("https://depmap.org/portal/download/api/download/external?file_name=ccle%2Fdepmap-mutation-calls-9a1a.7%2Fdepmap_19Q1_mutation_calls.csv")
depmap_19Q1_mutation_calls

# download public_19Q1_gene_cn.csv
public_19Q1_gene_cn  <- read.csv("https://depmap.org/portal/download/api/download/external?file_name=ccle%2Fdepmap-wes-cn-data-97cc.14%2Fpublic_19Q1_gene_cn.csv")
public_19Q1_gene_cn

# download DepMap-2019q1-celllines.csv
depmap_19Q1_cell_lines  <- read.csv("https://depmap.org/portal/download/api/download/external?file_name=processed_portal_downloads%2Fdepmap-public-cell-line-metadata-183e.3%2FDepMap-2019q1-celllines.csv")
depmap_19Q1_cell_lines

# download gene_effect_corrected.csv
gene_effect_corrected <- read.csv("https://ndownloader.figshare.com/files/14221385")
gene_effect_corrected

# download CCLE_depMap_19Q1_TPM.csv
CCLE_depMap_19Q1_TPM  <- read.csv("https://depmap.org/portal/download/api/download/external?file_name=ccle%2Fdepmap-rnaseq-expression-data-ccd0.12%2FCCLE_depMap_19Q1_TPM.csv")
CCLE_depMap_19Q1_TPM

# download CCLE_RPPA_20180123.csv
CCLE_RPPA_20180123 <- read.csv("https://depmap.org/portal/download/api/download/external?file_name=ccle%2FCCLE_RPPA_20180123.csv")
CCLE_RPPA_20180123

# download D2_combined_genetic_dependency_scores.csv
D2_combined_genetic_dependency_scores <- read.csv("https://ndownloader.figshare.com/files/13515395")
D2_combined_genetic_dependency_scores
```

Here, we have tried to preserve the original names of the
[depmap](https://github.com/tfkillian/depmap/) .csv files, however many of these 
names are long, unwieldy and don't immediately give the user the accurate 
impression of what is contained in the data frame. We will give these datasets 
new names to address this. 

Each quarter, the Broad Institute will release new depmap datasets. It is 
anticipated that the datasets will change as additional data comes out. For 
instance, the object `mutationCalls` reflects the most current depmap dataset. 
Older datasets such as `depmap_19Q1_mutation_calls` can be loaded after 
downloading the .csv directly from the 
[Broad Institute](https://depmap.org/portal/download/) portal.

In the following chucks, data cleaning of the data obtained from the 
[Broad Institute](https://depmap.org/portal/download/) portal will be performed.
However, please bear in mind that the directly downloaded .csv files from the 
[Broad Institute](https://depmap.org/portal/download/) portal will not feature
any of these data cleaning steps shown below. However, a moderately experienced
R user will be able to reproduce these steps on any depmap dataset given the 
code found in the chunks in the vignettes shown below.

```{r, eval=FALSE, echo=T}
# rename all the depmap dataframes to more convenient names

# rename 'D2_combined_genetic_dependency_scores' to 'rnai' data
rnai <- D2_combined_genetic_dependency_scores

# rename 'gene_effect_corrected' to 'crispr' data
crispr <- gene_effect_corrected

# rename 'depmap_19Q1_mutation_calls' to 'rnai' mutationCalls
mutationCalls <- depmap_19Q1_mutation_calls

# rename 'depmap_19Q1_cell_lines' to 'metadata' data
metadata <- depmap_19Q1_cell_lines

# rename 'public_19Q1_gene_cn' to 'copyNumber' data
copyNumber <- public_19Q1_gene_cn

# rename 'CCLE_RPPA_20180123' to 'RPPA' data
RPPA <- CCLE_RPPA_20180123

# rename 'CCLE_depMap_19Q1_TPM' to 'TPM' data
TPM <- CCLE_depMap_19Q1_TPM
```

The 'D2_combined_genetic_dependency_scores' a.k.a. 'rnai' dataset has common
gene names as the row names and the cell lines as the columns, unlike the other
datasets. This dataset needs to be transposed to make to it conform to the other
datasets.

```{r, eval=FALSE, echo=T}
#load necessary libraries
library(dplyr, warn.conflicts = FALSE)

#transpose D2_combined_genetic_dependency_scores
tranD2 <- t(D2_combined_genetic_dependency_scores) 

#coerce into df because tranD2 is currently a matrix
tranD2 <- as.data.frame(tranD2, row.names = NULL, col.names = TRUE) 

#select first column of tranD2, make it the colnames
colnames(tranD2) <- as.character(unlist(tranD2[1,]))

#drop first row (redundant with colnames(tranD2))
tranD2 = tranD2[-1, ] 

# we need the depmap ID  from `depmap_19Q1_cell_lines`
depIDtoName <- data.frame(depmap_19Q1_cell_lines$DepMap_ID, 
                       depmap_19Q1_cell_lines$CCLE_Name)

#add col names to depIDtoName to make the variables more clear
colnames(depIDtoName, do.NULL = FALSE)
colnames(depIDtoName) <- c("depmapID","cellLine")

# make new data frame with cellToID2, with DepmapID as a colname at the 
# beginning and the rest of the data frame on the columns to the right. 
# We need to add row names (cell lines) and then merge

cellLinesD2 <- colnames(D2_combined_genetic_dependency_scores)
cellLinesD2 <- as.data.frame(cellLinesD2)

# drop X1 row (not necessary) to make dimensions match
cellLinesD2 = cellLinesD2[-1, ] 

# merge cell lines with transposed D2 dataframe
cellToID2 <- cbind(cellLinesD2, tranD2)

# Rename first column to "cellLine" to it can be merged with `metadata`
names(cellToID2)[1] <-"cellLine"

# map DepmapID onto CCLE name for cellToID2 (transpose of 
# D2_combined_genetic_dependency_scores)
cellToID3 <- full_join(cellToID2, depIDtoName, all=TRUE)

# tidy up dataframe to make depmapID appear as the first column
depmapID <- cellToID3$depmapID
cellToIdCols <- subset(cellToID3, select = -c(depmapID))
cellToID4 <- cbind(depmapID, cellToIdCols)

# remove all NAs from column cellToID4$depmapID and rename to `noNAdepIDrnai`
noNAdepIDrnai <- subset(cellToID4, !is.na(depmapID))

# convert our cleaned data back to the rnai object
rnai <- noNAdepIDrnai
```

Some of these datasets do not have DepMap ID clearly labelled in some datasets,
instead the column is named (for instance) 'X1'. We will clean up the data here 
and provide column names which are common to all dataframes.

```{r, eval=FALSE, echo=T}
# Rename first column of `crispr` to "depmapID" 
names(crispr)[1] <-"depmapID"

# Rename first column of `mutationCalls` to "ID" 
names(mutationCalls)[1] <-"ID"

# change metadata$CCLE_Name to cellLine
names(metadata)[2] <- "cellLine"

# Rename first column of `copyNumber` to "depmapID" 
names(copyNumber)[1] <-"depmapID"

# Rename first column of `RPPA` to "cellLine" 
names(RPPA)[1] <-"cellLine"

# Rename first column of `copyNumber` to "depmapID" 
names(TPM)[1] <-"depmapID"
```

While name improperly named columns have been renamed, not all depmap datasets
possess both "depmapID" or "cellLine". We will be performing joins and data 
transformations such that every dataset (except metadata) contains "depmapID" as
the first column and "cellLine" as the second column.  

Note: If you are performing a join yourself, remember that this operation is 
computationally expensive and will take a bit to perform. This is why we are 
performing the set here so our end users of depmap don't have to!

```{r, eval=FALSE, echo=T}
# add 'cellLine' to 'crispr'
cripsrJoin <- full_join(crispr, depIDtoName, all=TRUE)

# add 'cellLine' to 'copyNumber'
cnJoin <- full_join(copyNumber, depIDtoName, all=TRUE)

# add 'depmapID' to 'RPPA'
rppaJoin <- full_join(RPPA, depIDtoName, all=TRUE)

# add 'cellLine' to 'TPM'
tpmJoin <- full_join(TPM, depIDtoName, all=TRUE)
```

Clean up to make sure that "depmapID" is the first column and "cellLine" is the 
second column

```{r, eval=FALSE, echo=T}
# rearrange rows to "depmapID" as first column and "cellLine" as second column

# for crisprJoin (crispr)
crisprDepID <- as.data.frame(cripsrJoin$depmapID)
crisprCell <- as.data.frame(cripsrJoin$cellLine)
cripsrJoinCols <- subset(cripsrJoin, select = -c(depmapID, cellLine))
df1 <- cbind(crisprDepID, crisprCell, cripsrJoinCols)
# column names need to be changed
names(df1)[1] <-"depmapID"
names(df1)[2] <-"cellLine"
crispr <- df1

# for cnJoin (copyNumber)
cnDepID <- as.data.frame(cnJoin$depmapID)
cnCell <- as.data.frame(cnJoin$cellLine)
cnJoinCols <- subset(cnJoin, select = -c(depmapID, cellLine))
df2 <- cbind(cnDepID, cnCell, cnJoinCols)
# column names need to be changed
names(df2)[1] <-"depmapID"
names(df2)[2] <-"cellLine"
copyNumber <- df2

# for rppaJoin (RPPA)
rppaDepID <- rppaJoin$depmapID
rppaJoinCols <- subset(rppaJoin, select = -c(depmapID))
df3 <- cbind(rppaDepID, rppaJoinCols)
# column names need to be changed
names(df3)[1] <-"depmapID"
RPPA <- df3

# for tpmJoin (TPM)
tpmDepID <- tpmJoin$depmapID
tpmCell <- tpmJoin$cellLine
tpmJoinCols <- subset(tpmJoin, select = -c(depmapID, cellLine))
df4 <- cbind(tpmDepID, tpmCell, tpmJoinCols)
# column names need to be changed
names(df4)[1] <-"depmapID"
names(df4)[2] <-"cellLine"
TPM <- df4
```

Rows with NA value for "depmapID" will be removed, to preserve the original size 
of the dataframe.

```{r, eval=FALSE, echo=T}
# make sure that there are now rows with NA in depmapID

# remove all NAs from column TPM$depmapID 
TPM <- subset(TPM, !is.na(depmapID))

# remove all NAs from column RPPA$depmapID 
RPPA <- subset(RPPA, !is.na(depmapID))

# remove all NAs from column copyNumber$depmapID 
copyNumber <- subset(copyNumber, !is.na(depmapID))

# remove all NAs from column rnai$depmapID 
rnai <- subset(rnai, !is.na(depmapID))

# remove all NAs from column crispr$depmapID 
crispr <- subset(crispr, !is.na(depmapID))

# manually view datasets to ensure that they are of the correct dimensions

# rnai
View(rnai[1:10, 1:10]) 
dim(rnai)
# [1]  1677 17311

# crispr
View(crispr[1:10, 1:10])
dim(crispr)
# [1]  1677 17636

# mutationCalls
View(mutationCalls[1:10, 1:10]) 
dim(mutationCalls)
# [1] 1243145      36

# copyNumber
View(copyNumber[1:10, 1:10])
dim(copyNumber)
# [1]  1678 23301

# RPPA
View(RPPA[1:10, 1:10])
dim(RPPA)
# [1] 1677  216

# TPM
View(TPM[1:10, 1:10])
dim(TPM)
# [1]  1677 57822

# metadata
View(metadata[1:10,])
dim(metadata)
#[1] 1677    9
```

In the last step, the cleaned depmap data was loaded into the depmap package
using devtools::use_data()

```{r, eval=FALSE, echo=T}

# include cleaned depmap data in depmap R package
devtools::use_data(rnai)
devtools::use_data(crispr)
devtools::use_data(TPM)
devtools::use_data(RPPA)
devtools::use_data(metadata)
devtools::use_data(mutationCalls)
devtools::use_data(copyNumber)
```