---
title: "depmap vignette"
author: "Theo Killian"
date: "February 19, 2019"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{depmap vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\usepackage[utf8]{inputenc}
---

```{r, include = FALSE, echo = FALSE, results = 'asis'}
knitr::opts_chunk$set(collapse = T, comment = "#>")
```

# [depmap](https://github.com/tfkillian/depmap/)

This is a brief demo vignette displaying the data and functions found in 
the [depmap](https://github.com/tfkillian/depmap/) package.  

## Introduction {#sec:intro}

The [depmap](https://github.com/tfkillian/depmap/) package aims to provide a 
reproducible research framework to cancer dependency data described by 
[Tsherniak, Aviad, et al. "Defining a cancer dependency map." Cell 170.3 (2017): 564-576.](https://www.ncbi.nlm.nih.gov/pubmed/28753430). The 
[depmap](https://github.com/tfkillian/depmap/) package should allow a researcher
to easily mine dependency data taken from this cancer genomic study, explore and
mine the data.

```{r, echo=FALSE, warning=FALSE}
# This code chunk simply makes sure that all the libraries used here are 
# installed, it will not be shown in the report (notice echo = FALSE).

packages <- c("knitr", "tidyr", "dplyr", "ggplot2", "plotly", "viridis", 
              "tidyverse", "grid", "gridExtra", "reshape2")
if ( length(missing_pkgs <- setdiff(packages, 
                                    rownames(installed.packages()))) > 0) {
  message("Installing missing package(s): ", 
          paste(missing_pkgs, collapse = ", "))
  install.packages(missing_pkgs)
}
```

## Preliminaries

To install [depmap](https://github.com/tfkillian/depmap/), the [BiocManager](https://cran.r-project.org/web/packages/BiocManager/index.html) 
Bioconductor Project Package Manager is required. If [BiocManager](https://cran.r-project.org/web/packages/BiocManager/index.html) 
is not already installed, it will need to be done so beforehand. Type (within R)
install.packages("BiocManager") (This needs to be done just once.)

```{r, eval=FALSE, echo=TRUE}
install.packages("BiocManager")
BiocManager::install("tfkillian/depmap")
```

## Data Import {#sec:data}

The [depmap](https://github.com/tfkillian/depmap/) package currently contains 
seven datasets (shown below) in *.rda* format. These datasets can be loaded by 
using *data()* after the [depmap](https://github.com/tfkillian/depmap/) package 
has been loaded in RStudio using *library("depmap")*. If you wish to see how the
data was transformed, please see the dataCleaning.Rmd vignette. 

```{r, eval=FALSE, echo=TRUE}
library("depmap")
#load(package_version())
```

Brief views of [depmap](https://github.com/tfkillian/depmap/) column subsets are
shown. Some of these datasets possess thousands of columns, which may be gene
names or in other cases cell lineages. For convenience, the list of column names
of these large data sets is restricted to the first 10. 

### "rnai" 
"rnai" is combined genetic Dependency Combined RNAi data taken from (Broad, 
Novartis, Marcotte). This dataset contains 17309 Genes, 712 Cell Lines, 30 
Primary Diseases, 31 Lineages. First column is the DepMap ID# of the cell 
lineage and the second column is the CCLE name for that cell line. The remaning 
columns are the dependency scores of the knockout of select genes in specific 
cancer cell lineages. 

```{r, eval=FALSE, echo=TRUE}
data("rnai", package = "depmap")
rnai
print("rnai data subset column names 1-10")
names(rnai[c(1:10)])
# [1] "rnai data subset column names 1-10"
# [1] "depmapID"        "cellLine"        "A1BG (1)"        "NAT2 (10)"      
# [5] "ADA (100)"       "CDH2 (1000)"     "AKT3 (10000)"    "MED6 (10001)"   
# [9] "NR2E3 (10002)"   "NAALAD2 (10003)"
# View(rnai[c(1:10, 1:10)])
# dim(rnai)
# [1]   709 17311
```

### "crispr"
"crispr" is the batch corrected CERES inferred gene effect dataset derived from 
CRISPR-Cas9 knockout of select genes in various cancer cell lines. This dataset 
contains 17634 Genes, 558 Cell Lines, 26 Primary Diseases, 28 Lineages. The 
first column is the DepMap ID# of the cell lineage and the second column is the 
CCLE name for that cell line. The remaining columns consist of the numerical 
dependency score values of the select genes. 

```{r, eval=FALSE, echo=TRUE}
data("crispr", package = "depmap")
crispr
print("crispr data subset column names 1-10")
names(crispr[c(1:10)])
# [1] "crispr data subset column names 1-10"
# [1] "depmapID"         "cellLine"         "A1BG (1)"        
# [4] "A1CF (29974)"     "A2M (2)"          "A2ML1 (144568)"  
# [7] "A3GALT2 (127550)" "A4GALT (53947)"   "A4GNT (51146)"   
# [10] "AAAS (8086)"  
# View(crispr[c(1:10, 1:10)])
# dim(crispr)
# [1]   558 17635
```

### "copyNumber" 
"copyNumber" is the depMap WES copy number data. This dataset contains 23299 
Genes, 1604 Cell Lines, 38 Primary Diseases, 33 Lineages. The first column is 
the DepMap ID# of the cell lineage and the second column is the CCLE name for 
that cell line. The remaining columns consist of the numerical log-fold copy 
number change measured against copy number baseline of select genes. 

```{r, eval=FALSE, echo=TRUE}
data("copyNumber", package = "depmap")
copyNumber
print("copyNumber data subset column names 1-10")
names(copyNumber[c(1:10)])
# [1] "copyNumber data subset column names 1-10"
# [1] "depmapID"               "cellLine"               "A1BG (1)"              
# [4] "NAT2 (10)"              "ADA (100)"              "CDH2 (1000)"           
# [7] "AKT3 (10000)"           "GAGE12F (100008586)"    "ZBTB11-AS1 (100009676)"
# [10] "MED6 (10001)"         
# View(copyNumber[c(1:10, 1:10)])
# dim(copyNumber)
# [1]  1604 23300
```

### "TPM"
"TPM"" dataset is the CCLE RNAseq gene expression data (log2(TPM+1)). This shows
expression data only for protein coding genes. This dataset contains: 57820 
Genes, 1165 Cell Lines, 33 Primary Diseases, 32 Lineages. First column is the 
DepMap ID# of the cell lineage and the second column is the CCLE name for that 
cell line. The remaining columns consist of the gene expression data.

```{r, eval=FALSE, echo=TRUE}
data("TPM", package = "depmap")
TPM
print("TPM data subset column names 1-10")
names((TPM[c(1:10)]))
# [1] "TPM data subset column names 1-10"
# [1] "depmapID"                   "cellLine"                  
# [3] "TSPAN6 (ENSG00000000003)"   "TNMD (ENSG00000000005)"    
# [5] "DPM1 (ENSG00000000419)"     "SCYL3 (ENSG00000000457)"   
# [7] "C1orf112 (ENSG00000000460)" "FGR (ENSG00000000938)"     
# [9] "CFH (ENSG00000000971)"      "FUCA2 (ENSG00000001036)"   
# View(TPM[c(1:10, 1:10)])
# dim(TPM)
# [1]  1165 57821
```

### "RPPA"
"RPPA" is the" CCLE Reverse Phase Protein Array (RPPA) dataset which contains: 
214 Genes, 899 Cell Lines, 28 Primary Diseases, 28 Lineages. First column is the
DepMap ID# of the cell lineage and the second column is the CCLE name for that 
cell line. The remaining columns consist of the numerical RPPA data of select 
genes. 

```{r, eval=FALSE, echo=TRUE}
data("RPPA", package = "depmap")
RPPA
print("RPPA data subset column names 1-10")
names((RPPA[c(1:10)]))
# [1] "RPPA data subset column names 1-10"
# [1] "depmapID"               "cellLine"              
# [3] "14-3-3_beta"            "14-3-3_epsilon_Caution"
# [5] "14-3-3_zeta"            "4E-BP1"                
# [7] "4E-BP1_pS65"            "4E-BP1_pT37_T46"       
# [9] "4E-BP1_pT70"            "53BP1"        
# View(RPPA[c(1:10, 1:10)])
# dim(RPPA)
# [1] 899 215
```

### "metadata" 
"metadata" dataset is the metadata about the cancer cell lines in the 19Q1 
depmap release. This dataset contains: 0 Genes, 1677 Cell Lines, 38 Primary 
Diseases, 33 Lineages. The columns found in this dataset are "depmapID", 
"cellLine", "Aliases", "COSMIC_ID", "Sanger.ID", "Primary.Disease", 
"Subtype.Disease", "Gender" and "Source"  

```{r, eval=FALSE, echo=TRUE}
data("metadata", package = "depmap")
metadata
print("metadata data column names")
names(metadata)
# [1] "metadata data column names"
# [1] "depmapID"       "cellLine"        "Aliases"         "COSMIC_ID"      
# [5] "Sanger.ID"       "Primary.Disease" "Subtype.Disease" "Gender"         
# [9] "Source" 
# View(metadata)
# dim(metadata)
# [1] 1677    9
```

### "mutationCalls" 
"mutationCalls" refers to the merged mutation calls (coding region, germline 
filtered). This dataset contains: 18755 Genes, 1601 Cell Lines, 37 Primary 
Diseases, 33 Lineages. The columns of this dataset are: "depmapID", 
"Hugo_Symbol", "Entrez_Gene_Id", "NCBI_Build", "Chromosome", "Start_position",        
"End_position", "Strand", "Variant_Classification", "Variant_Type", 
"Reference_Allele", "Tumor_Seq_Allele1", "dbSNP_RS", "dbSNP_Val_Status", 
"Genome_Change", "Annotation_Transcript", "Tumor_Sample_Barcode", "cDNA_Change",
"Codon_Change", "Protein_Change", "isDeleterious", "isTCGAhotspot", "TCGAhsCnt",
"isCOSMIChotspot", "COSMIChsCnt", "ExAC_AF", "VA_WES_AC", "CGA_WES_AC", 
"SangerWES_AC", "SangerRecalibWES_AC", "RNAseq_AC", "HC_AC", "RD_AC", "WGS_AC", 
"Variant_annotation", "DepMap_ID"  

```{r, eval=FALSE, echo=TRUE}
data("mutationCalls", package = "depmap")
mutationCalls
print("mutationCalls data column names")
names(mutationCalls)
# [1] "mutationCalls data column names"
# [1] "ID"                     "Hugo_Symbol"           
# [3] "Entrez_Gene_Id"         "NCBI_Build"            
# [5] "Chromosome"             "Start_position"        
# [7] "End_position"           "Strand"                
# [9] "Variant_Classification" "Variant_Type"          
# [11] "Reference_Allele"       "Tumor_Seq_Allele1"     
# [13] "dbSNP_RS"               "dbSNP_Val_Status"      
# [15] "Genome_Change"          "Annotation_Transcript" 
# [17] "Tumor_Sample_Barcode"   "cDNA_Change"           
# [19] "Codon_Change"           "Protein_Change"        
# [21] "isDeleterious"          "isTCGAhotspot"         
# [23] "TCGAhsCnt"              "isCOSMIChotspot"       
# [25] "COSMIChsCnt"            "ExAC_AF"               
# [27] "VA_WES_AC"              "CGA_WES_AC"            
# [29] "SangerWES_AC"           "SangerRecalibWES_AC"   
# [31] "RNAseq_AC"              "HC_AC"                 
# [33] "RD_AC"                  "WGS_AC"                
# [35] "Variant_annotation"     "depmapID"     
# View(mutationCalls)
# dim(mutationCalls)
# [1] 1243145      36
```

## Downloading [Broad Institute](https://depmap.org/portal/download/) data 

If desired, the [depmap](https://github.com/tfkillian/depmap/) datasets can also 
be downloaded directly in *.csv* format directly from the 
[Broad Institute](https://depmap.org/portal/download/) website. The instructions
on how to do so can be found in the dataCleaning vignette. Note: these files are
1.5GB in total (in non-compressed *.csv* format) and take a moderate amount of 
time to download.

It should also be noted that some data cleaning has been performed on the 
original [Broad Institute](https://depmap.org/portal/download/) datasets such as
renaming of some variables (e.g. "DepMap_ID" to "depmapID" and the addition of 
"depmapId" or "cellLine" columns to datasets that did not have one of these 
variables. The vignette describing how the depmap data was cleaned for this 
package can be found here (insert URL).

Additionally, the original data that the depmap data is derived from (raw DRIVE 
data) is also available for download. The raw DRIVE data has not been processed
using the CERES or DEMETER2 dependency inference algorithms. This data can be 
downloaded from the following link if so desired:
https://data.mendeley.com/datasets/y3ds55n88r/4

## Visualisations

Below are a few plots to help illustrate features of the 
[depmap](https://github.com/tfkillian/depmap/) data. The data loaded in the 
[depmap](https://github.com/tfkillian/depmap/) package is taken from a diverse 
array of cancer lineages, intended to represent the full spectrum of cancer. 

```{r, eval=TRUE, echo=FALSE, fig.width=10}
library(ggplot2, warn.conflicts = FALSE) 
library(viridis, warn.conflicts = FALSE)

#select data columns
cancerTypes <-unique(metadata$Primary.Disease)
numCancer <-(as.integer(cancerTypes))

#plot pie
pie(x=numCancer, label = "", init.angle = 90, 
    col = plasma(length(cancerTypes)), 
    main = "Proportion of 501 Cancer Lineages Represented in DepMap")
legend("left", legend=cancerTypes, fill=plasma(length(cancerTypes)))
```

Barplots of cancer mutation types

```{r, eval=TRUE, echo=FALSE}
library(gridExtra)
library(grid)

# Barplot of cancer mutation types
p1 <- ggplot(mutationCalls, aes(x=factor(Variant_Classification)))+ 
    geom_bar(stat="Count", width=0.9, fill="steelblue")+theme_minimal() +
    ggtitle("Mutation class") 

# Horizontal bar plot
p2 <- (p1 + coord_flip() + theme_gray())

# Barplot of cancer mutation types (SNP)
p3 <- ggplot(mutationCalls, aes(x=factor(Variant_Type))) +
    geom_bar(stat="Count", width=0.7, fill="steelblue")+theme_minimal() +
    ggtitle("Mutations type") 

# Horizontal bar plot
p4 <- (p3 + coord_flip() + theme_gray())

#plot as 1x2 grid
grid.arrange(p2, p4, nrow = 1, top = "Depmap Cancer Mutations")
```

Much of the data that can be found within datasets of the the 
[depmap](https://github.com/tfkillian/depmap/) package relates to the dependency 
score. This score represents the a scored (derived from processing via machine 
learning algorithms such as CERES or DEMETER2) which observe the shRNA log fold
change (LFC) depletion values in each cell line (CL) after 16 generations after 
gene knockdown/knockout. This dependency score is intended to express how vital 
a particular gene is in terms of how lethal the knockout/knockdown of that gene 
is on a target cell line. A highly negative dependency score is indicative of 
high lethality and therefore implying that that cell line is highly dependent on
that gene. Whereas, a positive score indicates improved vitality of a given 
cancer cell line. A dependency score of zero would indicate little to no effect
on vitality of gene knockout/knockdown.   

We will demonstrate how to obtain single dependency scores corresponding to a
specific gene and cell lineage. For example, we would like to know what 
dependency is present for a well known human tumor suppressor gene, like `BRCA1`
when it is knocked down via RNAi in a breast cancer lineage such as 
`184A1_BREAST`. This data can be found in the `rnai` dataset. 

```{r, eval=TRUE, echo=TRUE}
#select row and column and print dependency score
dsBRCA1breast <- rnai$`BRCA1 (672)`[which(rnai$cellLine=="184A1_BREAST")]
BRCA1val <- paste("Dep score for BRCA1 on 184A1_BREAST CL: ", dsBRCA1breast)
print(BRCA1val)
# "Dependency score for is BRCA1 on 184A1_BREAST CL is:  1.441402e-02"

#avg gene dependency
brca1DSavg <- mean(na.omit(rnai$`BRCA1 (672)`))
brca1DSavgVal <- paste("Avg BRCA1 dep score for all cell lines: ", brca1DSavg)
print(brca1DSavgVal)
# [1] "Avg BRCA1 dep score for all cell lines:  -0.157957653751371"
```

If we would like to investigate a specific gene and obtain the highest and 
lowest dependency scores via RNAI knock down for `BRCA1` along with the cancer 
cell lines associated with those values:

```{r, eval=TRUE, echo=TRUE}
#sort dependencies scores of particular gene (BRCA1) by lowest (most important)
lowBRCA1vals <-rnai[order(rnai$`BRCA1 (672)`, decreasing=F)[1:10],]

#sort dependencies scores of particular gene (BRCA1) my highest (most important)
highBRCA1vals <-rnai[order(rnai$`BRCA1 (672)`, decreasing=T)[1:10],]

# cbind as df
lowBRCA1 <- cbind(lowBRCA1vals$cellLine, lowBRCA1vals$`BRCA1 (672)`)
highBRCA1 <- cbind(highBRCA1vals$cellLine, highBRCA1vals$`BRCA1 (672)`)

print("Dependency scores for least 10 dependent cell lines for BRCA1 knockout")
highBRCA1 
#[1] "Dependency scores for least 10 dependent cell lines for BRCA1"
#      [,1]                             [,2]            
# [1,] "COLO678_LARGE_INTESTINE"        "0.487047073406"
# [2,] "ME180_CERVIX"                   "0.385984728794"
# [3,] "NCIH441_LUNG"                   "0.328893752785"
# [4,] "SQ1_LUNG"                       "0.309676934191"
# [5,] "CAKI1_KIDNEY"                   "0.260905572242"
# [6,] "D341MED_CENTRAL_NERVOUS_SYSTEM" "0.251014543174"
# [7,] "CW2_LARGE_INTESTINE"            "0.243416457789"
# [8,] "NCIH2291_LUNG"                  "0.235890487173"
# [9,] "NCIH1838_LUNG"                  "0.233686468271"
# [10,] "SNUC1_LARGE_INTESTINE"          "0.232145267234"

print("Dependency scores for top 10 dependent cell lines for BRCA1 knockout")
lowBRCA1 
#[1] "Dependency scores for top 10 dependent cell lines for BRCA1"
#  [,1]                           [,2]             
# [1,] "SMSCTR_SOFT_TISSUE"           "-0.815179289727"
# [2,] "JHOS2_OVARY"                  "-0.797872793957"
# [3,] "OVCAR8_OVARY"                 "-0.770237364698"
# [4,] "HCC1806_BREAST"               "-0.725152639186"
# [5,] "KPNSI9S_AUTONOMIC_GANGLIA"    "-0.709283349083"
# [6,] "HCC202_BREAST"                "-0.699951441734"
# [7,] "KMRC2_KIDNEY"                 "-0.680812400649"
# [8,] "DMS53_LUNG"                   "-0.646703047947"
# [9,] "M059K_CENTRAL_NERVOUS_SYSTEM" "-0.639093264076"
# [10,] "MDAMB330_BREAST"              "-0.616680884067"
```

If we would like to obtain the highest and lowest dependency scores for a 
certain cell line (for example `184A1_BREAST`) along with the genes associated 
with those values:

```{r, eval=TRUE, echo=TRUE}
# need to remove the columns with characters in `rnai` before finding max/min
rnai0Chars <- subset(rnai, select = -c(depmapID, cellLine))

# select row of rnai "184A1_BREAST"
breastRow <-which(rnai$cellLine=="184A1_BREAST")

# find column and value for lowest dep scoring gene for cell line "184A1_BREAST"
breastRowMin <-which.min(rnai0Chars[breastRow,])
breastRowMinVal <-rnai0Chars[breastRow,breastRowMin]
print("Gene with highest dependency score for cell line 184A1_BREAST")
breastRowMin
breastRowMinVal
# [1] "Gene with highest dependency score for cell line 184A1_BREAST"
# SF3B1 (23451) 
#         4888 
# [1] -2.29743

# find column and value for lowest dep scoring gene for cell line "184A1_BREAST"
breastRowMax <-which.max(rnai0Chars[breastRow,])
breastRowMaxVal <-rnai0Chars[breastRow,breastRowMax]
print("Gene with lowest dependency score for cell line 184A1_BREAST")
breastRowMax
breastRowMaxVal
# [1] "Gene with lowest dependency score for cell line 184A1_BREAST"
# TADA2A (6871) 
#        13331 
# [1] 1.430731

# find average value for dependency score for cell line "184A1_BREAST"
breastDSavg <-rowMeans(rnai0Chars[breastRow,], na.rm = TRUE, dims = 1)
print("Average gene dependency score for cell line 184A1_BREAST")
breastDSavg
#           4 
# -0.03864402 
```

Below shows a table which returns the 10 most significant (cancer-depleting) 
genes and their dependency scores for the `rnai` data, along with corresponding
cell line for that value. 

```{r, eval=TRUE, echo=FALSE}
# load dplyr
library(dplyr, warn.conflicts = FALSE)

# get gene columns from `rnai` subset `rnai0Chars`
geneNamesRNAI <-as.data.frame(names(rnai0Chars))

# add ID to list of genes (to merge at a later step)
geneNamesRNAI <- tibble::rowid_to_column(geneNamesRNAI, "ID")

# add col names
colnames(geneNamesRNAI, do.NULL = FALSE)
colnames(geneNamesRNAI) <- c("ID", "Gene")

# Return column index (gene) of each row minimum (cell line)
rnaiMinCols <- (apply(rnai0Chars, 1, which.min))

# lowest dependency scores for each column (cell line) of `rnai`
lowDepScoresColRNAI <- apply(rnai0Chars, 1, function(x) {min(x, na.rm = TRUE)})

# coerce to dataframe
rnaiMinCols <- as.data.frame(rnaiMinCols)
lowDepScoresColRNAI <- as.data.frame(lowDepScoresColRNAI)

# cbind `lowDepScoresColRNAI` and `rnaiMinCols`
rnaiDepScoreMinAndColIndex <- cbind(lowDepScoresColRNAI, rnaiMinCols)

# add col names
colnames(rnaiDepScoreMinAndColIndex, do.NULL = FALSE)
colnames(rnaiDepScoreMinAndColIndex) <- c("DepScore", "ID")

#left_join on ID
depMinCellGene <- left_join(rnaiDepScoreMinAndColIndex, geneNamesRNAI, all=TRUE)

# cbind `rnai$cellLine` and `depMinCellGene`
rnaiDepScoreMinAndColIndex <- cbind(rnai$cellLine, depMinCellGene)

# remove superfluous ID column
rnaiDepScoreMinAndColIndex <-subset(rnaiDepScoreMinAndColIndex, select = -c(ID))

#rename first column
names(rnaiDepScoreMinAndColIndex)[1] <- "cellLine"

# sort by highest dependencies 
top25rnai<-rnaiDepScoreMinAndColIndex[order(rnaiDepScoreMinAndColIndex$DepScore, 
                                             decreasing=F)[1:10],]
top25rnai 

# if we want to do the same as above, but column-wise

# Return min row index
# rnaiMinRows <- (apply(rnai0Chars,1,which.min))

# lowest dependency scores for each column of `rnai`
# lowDepScoresRowRNAI <-apply(rnai0Chars, 2, function(x) {min(x, na.rm = TRUE)})
```

Below shows a table which returns the 10 least significant (most cancer-driving) 
dependency scores for genes in the `rnai` data, along with corresponding cell 
line for that value. It may be interesting to compare these values with those 
found from the previous table.

```{r, eval=TRUE, echo=FALSE}
# Return column index (gene) of each row max (cell line)
rnaiMaxCols <- (apply(rnai0Chars, 1, which.max))

# highest dependency scores for each column (cell line) of `rnai`
highDepScoresColRNAI <- apply(rnai0Chars, 1, function(x) {max(x, na.rm = TRUE)})

# coerce to dataframe
rnaiMaxols <- as.data.frame(rnaiMaxCols)
highDepScoresColRNAI <- as.data.frame(highDepScoresColRNAI)

# cbind `highDepScoresColRNAI` and `rnaiMaxCols`
rnaiDepScoreMaxAndColIndex <- cbind(highDepScoresColRNAI, rnaiMaxCols)

# add col names
colnames(rnaiDepScoreMaxAndColIndex, do.NULL = FALSE)
colnames(rnaiDepScoreMaxAndColIndex) <- c("DepScore", "ID")

#left_join on ID
depMaxCellGene <- left_join(rnaiDepScoreMaxAndColIndex, geneNamesRNAI, all=TRUE)

# cbind `rnai$cellLine` and `depMaxCellGene`
rnaiDepScoreMaxAndColIndex <- cbind(rnai$cellLine, depMaxCellGene)

# remove superfluous ID column
rnaiDepScoreMaxAndColIndex <-subset(rnaiDepScoreMaxAndColIndex, select = -c(ID))

#rename first column
names(rnaiDepScoreMaxAndColIndex)[1] <- "cellLine"

# sort by highest dependencies 
bot25rnai<-rnaiDepScoreMaxAndColIndex[order(rnaiDepScoreMaxAndColIndex$DepScore, 
                                             decreasing=TRUE)[1:10],]
bot25rnai
```

We will do the same for what we did for the `rnai` data shown above (finding the 
population of genes with highest and lowest dependency scores) for the `crispr` 
data, as shown below. 

```{r, eval=TRUE, echo=FALSE}
# need to remove the columns with characters in `crispr` before finding max/min
crispr0Chars <- subset(crispr, select = -c(depmapID, cellLine))

# get gene columns from `crispr` subset `crispr0Chars`
geneNamesCRISPR <-as.data.frame(names(crispr0Chars))

# add ID to list of genes (to merge at a later step)
geneNamesCRISPR <- tibble::rowid_to_column(geneNamesCRISPR, "ID")

# add col names
colnames(geneNamesCRISPR, do.NULL = FALSE)
colnames(geneNamesCRISPR) <- c("ID", "Gene")

# Return column index (gene) of each row minimum (cell line)
crisprMinCols <- (apply(crispr0Chars, 1, which.min))

# lowest dependency scores for each column (cell line) of `crispr`
lowDepScoresColCRISPR <- apply(crispr0Chars, 1, 
                               function(x) {min(x, na.rm = TRUE)})

# coerce to dataframe
crisprMinCols <- as.data.frame(crisprMinCols)
lowDepScoresColCRISPR <- as.data.frame(lowDepScoresColCRISPR)

############## CRISPR or crispr? 

# cbind `lowDepScoresColCRISPR` and `crisprMinCols`
crisprDepScoreMinAndColIndex <- cbind(lowDepScoresColCRISPR, crisprMinCols)

# add col names
colnames(rnaiDepScoreMinAndColIndex, do.NULL = FALSE)
colnames(rnaiDepScoreMinAndColIndex) <- c("DepScore", "ID")

#left_join on ID
depMinCellGene <- left_join(rnaiDepScoreMinAndColIndex, geneNamesRNAI, all=TRUE)

# cbind `rnai$cellLine` and `depMinCellGene`
rnaiDepScoreMinAndColIndex <- cbind(rnai$cellLine, depMinCellGene)

# remove superfluous ID column
rnaiDepScoreMinAndColIndex <-subset(rnaiDepScoreMinAndColIndex, select = -c(ID))

#rename first column
names(rnaiDepScoreMinAndColIndex)[1] <- "cellLine"

# sort by highest dependencies 
top25rnai<-rnaiDepScoreMinAndColIndex[order(rnaiDepScoreMinAndColIndex$DepScore, 
                                             decreasing=F)[1:10],]
top25rnai 
```

##########################################################################

Below are a series of additional plots to help illustrate structures in the 
data. 

```{r}
# boxplot of high and low dependency score 

View(mutationCalls[1:10,])

#rando 
datRando <- data.frame(replicate(10,sample(0:1,1000,rep=TRUE)))
datRando
datRando[[1]] <- NULL
datRando
```


```{r, eval=FALSE, echo=T}
library(tidyverse, warn.conflicts = FALSE)
require(reshape2, warn.conflicts = FALSE)

## p5 - plot of dependency scores of 184A1_Breast
# omit genes with NA dependency and coerce into data frame
breastNoNA <-as.data.frame(na.omit(t(rnai0Chars[breastRow,])))

# add an ID column
breastNoNA <- tibble::rowid_to_column(breastNoNA, "ID")
Breast_184A1 <-breastNoNA

# add col names
colnames(Breast_184A1, do.NULL = FALSE)
colnames(Breast_184A1) <- c("Gene","Dependency")

# p1 - plot gene dependency levels for 184A1_BREAST cell line
p5 <-ggplot(Breast_184A1) + 
    aes(Gene, Dependency, color = Breast_184A1$Dependency) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=breastDSavg, linetype="dashed", color = "black") +
    ggtitle("Avg. RNAi dep. scores for each gene for cell line 184A1_Breast")

# p6 - boxplot of dependency scores of first 10 genes in `rnai`
# subset first 10 columns of `rnai` to create `first10Rnai`
first10Rnai <- rnai[3:13]

p6 <- ggplot(data = melt(first10Rnai), aes(x=variable, y=value)) + 
    geom_boxplot(aes(fill=variable)) + 
    theme(axis.text.x = element_text(angle = 90, hjust = 1)) + 
    ggtitle("Boxplot of dependency scores of first 10 genes in `rnai` dataset")

## p7 - plot of the lowest dep. scores for each gene for `rnai` data 
# apply function to get mean dependency of each gene from all columns
rnaiColAvg <- apply(rnai[c(3:17311)], 2, function(x) {mean(x, na.rm = TRUE)})
rnaiColAvgVal <- mean(rnaiColAvg)
# -0.06833473

#coerce into data frame
depAllGenes <- as.data.frame(t(rnaiColAvg))
depAllGenes <- tibble::rowid_to_column(depAllGenes, "ID")
names(depAllGenes)

#plot gene dependency levels for all cell lines
p7 <-ggplot(depAllGenes) + 
    aes(ID, Dependency_Score, color = depAllGenes$Dependency) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=rnaiColAvgVal, linetype="dashed", color = "black") + 
    geom_hline(yintercept=df3_avg, linetype="dashed", color = "red") +
    labs(y = "Dep Score of Gene with Largest Dep Score", x = "Cell Line") + 
    ggtitle("Greatest RNAi dependency scores for each gene by cell line") +
    scale_linetype_manual(name = "Avg dep scores",values = c(2,2))

# I will add a legend to the lines when I have time

#plot as 1x2 grid
grid.arrange(p5, p6, p7, nrow = 1, top = "Depmap Data Visualizations")

```

It might be interesting to locate the highest scoring dependencies among all 
cell lines for the CRISPR dataset as well.

```{r, eval=FALSE, echo=T}
# function that finds highest dependency scores for each gene and cell line

lowDepScoresCRISPR <- apply(crispr[c(2:17635)], 1, function(x) {min(x, na.rm = TRUE)})

#lowDepGeneCRISPR <- apply(crispr[c(2:17635)], 1, function(x) {depmin <-which.min(x) crispr[depmin]})

# coerce into data frame
lowDepScoresCRISPR <- as.data.frame(lowDepScoresCRISPR)
lowDepGeneCRISPR <- as.data.frame(lowDepGeneCRISPR)

df4<-cbind(lowDepGeneCRISPR, lowDepScoresCRISPR)

colnames(df4, do.NULL = FALSE)
colnames(df4) <- c("Gene","Dependency_Score", "Num")

# sort by 10 highest dependencies 
top10CRISPR <-df4[order(df4$Dependency_Score, decreasing=F)[1:10],]
top10CRISPR
```

Below is a plot of the lowest dependency scores for each gene and CL for the 
`crispr` dataset. 
 
```{r, eval=FALSE, echo=T}
# apply function to get mean dependency from all columns
gEc_avg <- apply(crispr[c(2:17635)], 1, function(x) {mean(x, na.rm = TRUE)})

gEc_avg2 <- mean(gEc_avg)
# -0.06833473

# apply function to get mean HIGHEST dependency from all columns
df4_avg <- mean(df4$Dependency_Score)
# -2.198807

#coerce into data frame
df4 <- as.data.frame(df4)
#df3 <- tibble::rowid_to_column(df3, "ID")
#View(df3)
depAllGenesCRISPR <-df4 

#plot gene dependency levels for all cell lines
ggplot(depAllGenesCRISPR) + aes(Num, Dependency_Score, color = depAllGenesCRISPR$Dependency_Score) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=gEc_avg2, linetype="dashed", color = "black") + 
    geom_hline(yintercept=df4_avg, linetype="dashed", color = "red") +
    labs(y = "Dep Score of Gene with Largest Dep Score", x = "Cell Line") + 
    ggtitle("Greatest CRISPR dependency scores for each gene by cell line") +
    scale_linetype_manual(name = "Avg dep scores",values = c(2,2))

# I will add a legend to the lines when I have time
```

It might be insightful to compare the dependency scores of datasets derived from
RNAi knockout e.g. `rnai` and those utilizing CRISPR knockouts e.g. `crispr`. 

```{r, eval=TRUE, echo=F}
# data wrangling for `crispr`

# Rename first column to "cellLine" to it can be merged with `metadata`
names(crispr)[1]<-"depmapID"

# map CCLE name onto DepmapID for crispr
geneToID <- full_join(crispr, metadata, all=TRUE)

# tidy up dataframe to make cellLine appear as the second column
depmapID <- geneToID$depmapID
cellLine <- geneToID$cellLine
geneToIDCols <- subset(cellToID3, select = -c(depmapID, cellLine))
geneIDcrispr <-cbind(depmapID, cellLine, geneToID)
geneIDcrispr <- geneIDcrispr[,-3] #remove reduncent depmapID column
```

```{r, eval=TRUE, echo=F}
# in order to plot the `crispr` (CRISPR) and `rnai`(RNAi) datasets, we will have
# to make them comparable, which means removing any NAs or genes that were not 
# tested in on or the other dataset. We will 1) remove all rows with NA for 
# dependency score for the PIK3C3 gene and 2) remove all genes (columns) that do
# not occur in both datasets, then 3) full_join on the cell lines (depmapID) in 
# common for the CRISPR and RNAi for the PIK3C3 gene dependencies of both 
# datasets 

# compare `depmapID` column of `geneIDcrispr` and `noNAdepIDrnai` 

# since the dimensions for `geneIDcrispr` and `noNAdepIDrnai` are the same, we
# will full_join on `depmapID` and create a new dataframe `depPIK3C3` with 
# depenency scores that can be compared between RNAi CRISPR datasets. Note: we
# use "cbind.data.frame" to avoid turning the numeric values into factors

crisprPIK3C3 <-cbind.data.frame(geneIDcrispr$depmapID, geneIDcrispr$`PIK3C3 (5289)`)
rnaiPIK3C3 <-cbind.data.frame(noNAdepIDrnai$depmapID, noNAdepIDrnai$`PIK3C3 (5289)`, stringsAsFactors = FALSE)

#rename column names 

#add col names to crisprPIK3C3 to make the variables more clear
colnames(crisprPIK3C3, do.NULL = FALSE)
colnames(crisprPIK3C3) <- c("depmapID","crisprPIK3C3")

#add col names to rnaiPIK3C3 to make the variables more clear
colnames(rnaiPIK3C3, do.NULL = FALSE)
colnames(rnaiPIK3C3) <- c("depmapID","rnaiPIK3C3")

#full_join on depmapID
comparePIK3C3 <- full_join(crisprPIK3C3, rnaiPIK3C3, all=TRUE)

#set all NAs to 0 for comparePIK3C3 so the dataframe can be visualized

# comparePIK3C3$rnaiPIK3C3 has to be turned into character because it throws 
# an annoying warning "invalid factor level, NA generated"
comparePIK3C3$rnaiPIK3C3<- as.character(comparePIK3C3$rnaiPIK3C3)
comparePIK3C3[, 1:3][is.na(comparePIK3C3[, 1:3])] <- 0

# comparePIK3C3$rnaiPIK3C3 converted back to numeric. Note: this is an inelegant
# solution, need to find something that works better 
comparePIK3C3$rnaiPIK3C3<- as.numeric(comparePIK3C3$rnaiPIK3C3)

#convert comparePIK3C3 to depPIK3C3, and convert 0 values to NA
depPIK3C3 <- comparePIK3C3
depPIK3C3[depPIK3C3==0] <- NA
depPIK3C3

#Delete the rows associated with NA.
compDepPIK3C3<-depPIK3C3[complete.cases(depPIK3C3),]
#View(compDepPIK3C3)
#dim(compDepPIK3C3)
#[1] 402   3
# this means that there are only 402 cell lines where there are dependency 
# scores for both RNAI and CRISPR. we will plot them in the chunk below

# sort byDepIdPIK3C3 by depmapID, lowest to highest
byDepIdPIK3C3 <- compDepPIK3C3[order(compDepPIK3C3$depmapID, decreasing=F),]
```

Now that the data wrangling for `rnai` and 
`crispr` is complete, we will attempt to visualize the difference
in dependency scores between the RNAi and CRISPR datasets. First, comparison for 
the dependency two genes between the two datasets will be produced, then a 
global comparison between the two datasets will be demonstrated.

We will begin by illustrating a histogram of the dependency scores for a 
particular gene, comparing CRISPR and RNAI.

Scatterplot of the dependency scores for RNAi knockouts for PIK3C3 using ggplot2

```{r, eval=FALSE, echo=T}
# histogram of the dependency scores for a particular gene taken from data-
# wrangled `crispr` and `rnai` datasets. We will first be visualizing the PIK3C3
# gene, which is one of # the most dependent genes in either dataset. We will do
# this by combining the geneIDcrispr$`PIK3C3 (5289)` and 
# noNAdepIDrnai$`PIK3C3 (5289)` columns into a single dataframe and plotting 
# with ggplot2 

avgCRISPR <-mean(byDepIdPIK3C3$crisprPIK3C3)
avgRNAI <-mean(byDepIdPIK3C3$rnaiPIK3C3)

#plot RNAI gene dependency levels for PIK3C3 for all complete cell lines 
ggplot(byDepIdPIK3C3) + aes(depmapID, byDepIdPIK3C3$rnaiPIK3C3, color = byDepIdPIK3C3$rnaiPIK3C3) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=avgRNAI, linetype="dashed", color = "red") +
    labs(y = "Dependency Score", x = "Cell Lines") + 
    ggtitle("Plot of the dependency scores for RNAi for PIK3C3") 
```

Scatterplot of dependency scores for CRISPR knockouts for PIK3C3 using ggplot2

```{r, eval=FALSE, echo=T}
#plot CRISPR gene dependency levels for PIK3C3 for all complete cell lines
ggplot(byDepIdPIK3C3) + aes(depmapID, byDepIdPIK3C3$crisprPIK3C3, color = byDepIdPIK3C3$crisprPIK3C3) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=avgCRISPR, linetype="dashed", color = "blue") + 
    labs(y = "Dependency Score", x = "Cell Lines") + 
    ggtitle("Plot of the dependency scores for CRISPR for PIK3C3") 
```

Overlay of dependency scores for RNAi and CRISPR knockouts for PIK3C3 

```{r, eval=FALSE, echo=T}
#overlay RNAI and CRISPR
library(lattice)
xyplot(crisprPIK3C3 + rnaiPIK3C3 ~ sequence(nrow(byDepIdPIK3C3)), data=byDepIdPIK3C3, 
       pch = 21, fill = c("blue", "red"), cex = 1, xlab=list("Cell Lines", cex=1.7), 
       ylab=list("Dependency Score for Gene", cex=1.5), 
       main=list("Dependency Score of PIKC3C for RNAi and CRISPR for all cell lines", cex=1.1),
       auto.key=list(space = "top", column = 2, title="Knockout Type", cex.title=1),
       panel = function(x, y, ...)
       {panel.xyplot(x, y, ...)  ## First call the default panel function for 'xyplot'
       panel.abline(h = avgCRISPR, col = 2, lty = 2)
       panel.abline(h = avgRNAI, col = 1, lty = 2)
       })
```

Plot CRISPR vs RNAi gene dependency levels for PIK3C3

```{r, eval=FALSE, echo=T}
#plot CRISPR vs RNAI gene dependency levels for PIK3C3 for all complete cell lines 
ggplot(byDepIdPIK3C3) + aes(rnaiPIK3C3, crisprPIK3C3, color = byDepIdPIK3C3$rnaiPIK3C3) + 
    geom_point() +  scale_colour_viridis() + 
    labs(y = "CRISPR Dependency Score", x = "RNAI Dependency Score") + 
    ggtitle("Dependency scores for RNAi vs CRISPR for PIKC3C") 
```

Compare CRISPR vs RNAi for COPB1

```{r, eval=FALSE, echo=T}
# since the dimensions for `geneIDcrispr` and `noNAdepIDrnai` are the same, we
# will full_join on `depmapID` and create a new dataframe `depCOPB1` with 
# depenency scores that can be compared between RNAi CRISPR datasets. Note: we
# use "cbind.data.frame" to avoid turning the numeric values into factors

crisprCOPB1 <-cbind.data.frame(geneIDcrispr$depmapID, geneIDcrispr$`COPB1 (1315)`)
rnaiCOPB1 <-cbind.data.frame(noNAdepIDrnai$depmapID, noNAdepIDrnai$`COPB1 (1315)`, stringsAsFactors = FALSE)

#rename column names 

#add col names to crisprCOPB1 to make the variables more clear
colnames(crisprCOPB1, do.NULL = FALSE)
colnames(crisprCOPB1) <- c("depmapID","crisprCOPB1")

#add col names to rnaiCOPB1 to make the variables more clear
colnames(rnaiCOPB1, do.NULL = FALSE)
colnames(rnaiCOPB1) <- c("depmapID","rnaiCOPB1")

#full_join on depmapID
compareCOPB1 <- full_join(crisprCOPB1, rnaiCOPB1, all=TRUE)
#View(compareCOPB1)

compDepCOPB1<-compareCOPB1[complete.cases(compareCOPB1),]
#View(compDepCOPB1)

compDepCOPB1$rnaiCOPB1<- as.numeric(as.character(compDepCOPB1$rnaiCOPB1))

byDepIdCOPB1 <- compDepCOPB1
```

CRISPR vs RNAi for COPB1

```{r, eval=FALSE, echo=T}
#plot CRISPR vs RNAI gene dependency levels for COPB1 for all complete cell lines 
ggplot(byDepIdCOPB1) + aes(rnaiCOPB1, crisprCOPB1, color = byDepIdCOPB1$rnaiCOPB1) + 
    geom_point() +  scale_colour_viridis() + 
    labs(y = "CRISPR Dependency Score", x = "RNAI Dependency Score") + 
    ggtitle("Dependency scores for RNAi vs CRISPR for COPB1") 
```


Overlay of dependency scores for RNAi and CRISPR knockouts for COPB1 

```{r, eval=FALSE, echo=T}
avgCRISPR2 <-mean(byDepIdPIK3C3$crisprCOPB1)
avgRNAI2 <-mean(byDepIdPIK3C3$rnaiCOPB1)

#overlay RNAI and CRISPR
library(lattice)
xyplot(crisprCOPB1 + rnaiCOPB1 ~ sequence(nrow(byDepIdCOPB1)), data=byDepIdCOPB1, 
       pch = 21, fill = c("blue", "red"), cex = 1, xlab=list("Cell Lines", cex=1.7), 
       ylab=list("Dependency Score for Gene", cex=1.5), 
       main=list("Dependency Score of COPB1 for RNAi and CRISPR for all cell lines", cex=1.1),
       auto.key=list(space = "top", column = 2, title="Knockout Type", cex.title=1),
       panel = function(x, y, ...)
       {panel.xyplot(x, y, ...)  ## First call the default panel function for 'xyplot'
       panel.abline(h = avgCRISPR2, col = 2, lty = 2)
       panel.abline(h = avgRNAI2, col = 1, lty = 2)
       })
```

Some data wrangling must be performed to visualize the differences in dependency 
for both RNAi and CRISPR datasets for all genes. 

```{r, eval=TRUE, echo=F}
# apply function to calculate the mean of all of the dependencies for each gene

# convert geneIDcrispr to geneIDcrispr2 so we can preserve original df
geneIDcrispr2 <- geneIDcrispr

# convert noNAdepIDrnai to noNAdepIDrnai2 so we can preserve original df
noNAdepIDrnai2 <- noNAdepIDrnai

#nremove columns with characters
geneIDcrispr2sub <- geneIDcrispr2[3:17311]

crisprCellMean <- apply(geneIDcrispr2sub, 1, mean, na.rm = TRUE)

crisprGeneMean <- apply(geneIDcrispr2sub, 2, mean, na.rm = TRUE)

#remove columns with characters
noNAdepIDrnai2sub <- noNAdepIDrnai2[3:17311]

# the columns of noNAdepIDrnai2sub were converted to factors somehow, and we
# will be converting them back to numeric in order to plot or manipulate them
indx <- sapply(noNAdepIDrnai2sub, is.factor)
noNAdepIDrnai2sub[indx] <- lapply(noNAdepIDrnai2sub[indx], 
                                  function(x) as.numeric(as.character(x)))

rnaiCellMean <- apply(noNAdepIDrnai2sub, 1, mean, na.rm = TRUE)

rnaiGeneMean <- apply(noNAdepIDrnai2sub, 2, mean, na.rm = TRUE)

# now we need to make combined dataframes of CellMean and GeneMean
# (it's easier to manipulate and plot if they are in one dataframe)

# cbind the CellMeans
cellMeans <-cbind(crisprCellMean, rnaiCellMean)

#add col names to cellMeans to make the variables more clear
colnames(cellMeans, do.NULL = FALSE)
colnames(cellMeans) <- c("CRISPRavgCellDep","RNAIavgCellDep")

#coerce into data frame
cellMeans <- as.data.frame(cellMeans)

# add sequence number 
cellMeans <- tibble::rowid_to_column(cellMeans, "ID")

# cbind the GeneMeans
geneMeans <-cbind(crisprGeneMean, rnaiGeneMean)

#add col names to cellMeans to make the variables more clear
colnames(geneMeans, do.NULL = FALSE)
colnames(geneMeans) <- c("CRISPRavgGeneDep","RNAIavgGeneDep")

#coerce into data frame
geneMeans <- as.data.frame(geneMeans)

# add sequence number 
geneMeans <- tibble::rowid_to_column(geneMeans, "ID")

# there are a lot of missing averages in CellMeans (oddly enough), we will only
# visualise the complete cases

#Delete the rows associated with NA.
compCellMeans<-cellMeans[complete.cases(cellMeans),]

#Delete the rows associated with NA.
compGeneMeans<-geneMeans[complete.cases(geneMeans),]

# calculate averages to show in plots

# total average CRISPR Gene dependency for all genes
totalAvgCrisprGene <- mean(geneMeans$CRISPRavgGeneDep, na.rm = TRUE)

# total average RNAi Gene dependency for all genes
totalAvgRNAiGene <- mean(geneMeans$RNAIavgGeneDep, na.rm = TRUE)

# total average CRISPR Gene dependency for all cell lines
totalAvgCrisprCell <- mean(cellMeans$CRISPRavgCellDep, na.rm = TRUE)

# total average RNAi Gene dependency for all cell lines
totalAvgRNAiCell <- mean(cellMeans$RNAIavgCellDep, na.rm = TRUE)
```

Plot of avg CRISPR dep scores for all Cell Lines

```{r, eval=FALSE, echo=T}
#Plot of avg CRISPR dep scores for all genes
ggplot(compCellMeans) + aes(ID, CRISPRavgCellDep, color = CRISPRavgCellDep) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=totalAvgCrisprCell, linetype="dashed", color = "blue") + 
    labs(y = "Average dependency score per cell line", x = "Cell Lines") + 
    ggtitle("Average CRISPR dependency scores for all Cell Lines") 
```

Plot of average RNAI dependency scores for all cell lines

```{r, eval=FALSE, echo=T}
#Plot of avg RNAI dep scores for all genes
ggplot(compCellMeans) + aes(ID, RNAIavgCellDep, color = RNAIavgCellDep) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=totalAvgRNAiCell, linetype="dashed", color = "blue") + 
    labs(y = "Average dependency score per cell line", x = "Cell Lines") + 
    ggtitle("Average RNAI dependency scores for all cell lines") 
```

Plot of average CRISPR dependency scores for all genes

```{r, eval=FALSE, echo=T}
#Plot of avg RNAI dep scores for all genes
ggplot(compGeneMeans) + aes(ID, CRISPRavgGeneDep, color = CRISPRavgGeneDep) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=totalAvgCrisprGene, linetype="dashed", color = "blue") + 
    labs(y = "Average dependency score per gene", x = "Genes") + 
    ggtitle("Average CRISPR dependency scores for all genes") 
```

Plot of avg RNAI dep scores for all genes

```{r, eval=FALSE, echo=T}
#Plot of avg RNAI dep scores for all cell lines
ggplot(compGeneMeans) + aes(ID, RNAIavgGeneDep, color = RNAIavgGeneDep) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=totalAvgRNAiGene, linetype="dashed", color = "blue") + 
    labs(y = "Dependency Score", x = "Genes") + 
    ggtitle("Average RNAi dependency scores for all genes") 
```


Plot CRISPR vs RNAi average gene dependency levels all cell lines

```{r, eval=FALSE, echo=T}
#plot CRISPR vs RNAi average gene dependency levels all genes
p2 <- ggplot(compGeneMeans) + aes(CRISPRavgGeneDep, RNAIavgGeneDep, color = CRISPRavgGeneDep) + 
    geom_point() +  scale_colour_viridis() + geom_smooth(method='lm',formula=y~x) +
    labs(y = "RNAIavgGeneDep", x = "CRISPRavgGeneDep") + 
    ggtitle("CRISPR vs RNAi average gene dependency levels all genes") 
p2 + geom_abline(intercept = 0, slope = 1)
```

Plot CRISPR vs RNAi average gene dependency levels all cells

```{r, eval=FALSE, echo=T}
#plot CRISPR vs RNAi average gene dependency levels all cells
p1 <- ggplot(compCellMeans) + aes(CRISPRavgCellDep, RNAIavgCellDep, color = CRISPRavgCellDep) + 
    geom_point() +  scale_colour_viridis() + geom_smooth(method='lm',formula=y~x) +
    labs(y = "RNAIavgCellDep", x = "CRISPRavgCellDep") + 
    ggtitle("CRISPR vs RNAi average gene dependency levels all cell lines") 
p1 + geom_abline(intercept = 0, slope = 1)
```

Average Dependency Scores for RNAi and CRISPR for all genes

```{r, eval=FALSE, echo=T}
#overlay RNAI and CRISPR dependency scores for genes
xyplot(CRISPRavgGeneDep + RNAIavgGeneDep ~ sequence(nrow(compGeneMeans)), data=compGeneMeans, 
       pch = 21, fill = c("blue", "red"), cex = 1, xlab=list("Genes", cex=1.7), 
       ylab=list("Avg Dep Score for each gene", cex=1.5), 
       main=list("Average Dependency Scores for RNAi and CRISPR for all genes", cex=1.1),
       auto.key=list(space = "top", column = 2, title="Knockout Type", cex.title=1),
       panel = function(x, y, ...)
       {panel.xyplot(x, y, ...)  ## First call the default panel function for 'xyplot'
       panel.abline(h = totalAvgCrisprGene, col = 2, lty = 2)
       panel.abline(h = totalAvgRNAiGene, col = 1, lty = 2)
       })
```

Average Dependency Scores for RNAi and CRISPR for all cell lines

```{r, eval=FALSE, echo=T}
#overlay RNAI and CRISPR for genes
xyplot(CRISPRavgCellDep + RNAIavgCellDep ~ sequence(nrow(compCellMeans)), data=compCellMeans, 
       pch = 21, fill = c("blue", "red"), cex = 1, xlab=list("Cell Lines", cex=1.7), 
       ylab=list("Avg Dep Score for each cell line", cex=1.5), 
       main=list("Average Dependency Scores for RNAi and CRISPR for all cell lines", cex=1.1),
       auto.key=list(space = "top", column = 2, title="Knockout Type", cex.title=1),
       panel = function(x, y, ...)
       {panel.xyplot(x, y, ...)  ## First call the default panel function for 'xyplot'
       panel.abline(h = totalAvgCrisprCell, col = 2, lty = 2)
       panel.abline(h = totalAvgRNAiCell, col = 1, lty = 2)
       })
```

(4) explore the dependency scores for genes with high and low expression 

PSMC2 (Rpt1) was the highest-ranked CYCLOPS candidate gene in the original 
depmap analysis and was also significant in the validation data set. PSMC2 is 
part of the 19S regulatory complex of the 26S proteasome, which is responsible 
for catalyzing the unfolding and translocation of substrates into the 20S 
proteasome

Copy Number Note: Find out what the values of copy number are in copyNumber

```{r, eval=TRUE, echo=F}

# apply function to calculate the mean of all of the dependencies for each gene

#nremove columns with characters
CNsub <- copyNumber[3:23300]

copyNumGeneMean <- apply(CNsub, 1, mean, na.rm = TRUE)

copyNumCellMean <- apply(CNsub, 2, mean, na.rm = TRUE)

#coerce into data frame
copyNumGeneMean <- as.data.frame(copyNumGeneMean)
copyNumCellMean <- as.data.frame(copyNumCellMean)

# add sequence number 
copyNumGeneMean <- tibble::rowid_to_column(copyNumGeneMean, "ID")
copyNumCellMean <- tibble::rowid_to_column(copyNumCellMean, "ID")

#add col names to cellMeans to make the variables more clear
colnames(copyNumCellMean, do.NULL = FALSE)
colnames(copyNumCellMean) <- c("ID","copyNumCellMean")

#add col names to cellMeans to make the variables more clear
colnames(copyNumGeneMean, do.NULL = FALSE)
colnames(copyNumGeneMean) <- c("ID","copyNumGeneMean")

#Delete the rows associated with NA.
copyNumGeneMean<-geneMeans[complete.cases(copyNumGeneMean),]
copyNumCellMean<-geneMeans[complete.cases(copyNumCellMean),]

# calculate averages to show in plots

# total average CRISPR Gene dependency for all genes
totalAvgCNGene <- mean(copyNumGeneMean$copyNumGeneMean, na.rm = TRUE)
#totalAvgCrisprGene 

# total average RNAi Gene dependency for all genes
totalAvgCNCell <- mean(copyNumCellMean$copyNumCellMean, na.rm = TRUE)
```

Plot of avg CRISPR dep scores for all Cell Lines

```{r, eval=FALSE, echo=T}
#Plot of avg CRISPR dep scores for all genes
ggplot(compCellMeans) + aes(ID, CRISPRavgCellDep, color = CRISPRavgCellDep) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=totalAvgCrisprCell, linetype="dashed", color = "blue")
    + labs(y = "Average dependency score per cell line", x = "Cell Lines") + 
    ggtitle("Average CRISPR dependency scores for all Cell Lines") 
```

Scatterplot overlay

```{r, eval=FALSE, echo=T}
#combine two scatterplots in one
ggplot(df) + geom_point(aes(x = x1, y = y1)) + geom_point(aes(x = x2, y = y2)
```

############## TO FIX ############################

Here we will locate the highest scoring gene dependencies among all cell lines 
for the RNAI dataset. These dependencies might make for interesting targets for 
research.

```{r, eval=FALSE, echo=T}
# remove character columns of `rnai` dataset rename to `rnai0Chars`
rnai0Chars <- subset(rnai, select = -c(depmapID, cellLine))

#add row names so that they can be retrieved later
row.names(rnai0Chars) <- rnai$cellLine

# apply function finds lowest value dep. scores for each gene (column)
lowDepScores <- apply(rnai0Chars, 1, function(x) {min(x, na.rm = TRUE)})

# apply function finds (row) of lowest value dep. scores for each gene (column)
lowDepCellLine <- apply(rnai[c(3:17311)], 1, function(x) which.min(x))
lowDepCellLine <- as.data.frame(lowDepCellLine, row.names = FALSE)
colnames(lowDepCellLine, do.NULL = FALSE)
colnames(lowDepCellLine) <- c("cellLine")
View(lowDepCellLine)

lowDepCellLineList <- rnai$depmapID[lowDepCellLine,]

View(lowDepCellLineList[c(1:10)])

# coerce into data frame
lowDepScores <- as.data.frame(lowDepScores)
View(lowDepScores)
#lowDepCellLine <- as.data.frame(lowDepCellLine)

# turn row names into a column
r1 <-rownames(lowDepScores)
lowDepScores$Cell_Line <- r1
r2 <-rownames(lowDepCellLine)
lowDepCellLine$Cell_Line <- r2

# tidy up df, merge on Cell_Line
df3 = merge(lowDepCellLine, lowDepScores, by.x=c("Cell_Line"), by.y=c("Cell_Line"))
colnames(df3, do.NULL = FALSE)
colnames(df3) <- c("Cell_Line", "Gene", "Dependency_Score")

# sort by 10 highest dependencies 
top10 <-df3[order(df3$Dependency_Score, decreasing=F)[1:10],]
top10
```

It might be interesting to locate the highest scoring dependencies among all 
cell lines for the CRISPR dataset as well.

```{r, eval=FALSE, echo=T}
# function that finds highest dependency scores for each gene and cell line

lowDepScoresCRISPR <- apply(crispr[c(2:17635)], 1, function(x) {min(x, na.rm = TRUE)})

lowDepGeneCRISPR <- apply(crispr[c(2:17635)], 1, function(x) {depmin <-which.min(x) crispr$depmapID[depmin]})

# coerce into data frame
lowDepScoresCRISPR <- as.data.frame(lowDepScoresCRISPR)
lowDepGeneCRISPR <- as.data.frame(lowDepGeneCRISPR)

df4<-cbind(lowDepGeneCRISPR, lowDepScoresCRISPR)

colnames(df4, do.NULL = FALSE)
colnames(df4) <- c("Gene","Dependency_Score", "Num")

# sort by 10 highest dependencies 
top10CRISPR <-df4[order(df4$Dependency_Score, decreasing=F)[1:10],]
top10CRISPR
```

The gene with the highest dependency score is UBC (7316). This gene represents a 
ubiquitin gene, ubiquitin C. The encoded protein is a polyubiquitin precursor.  
Ubiquitination has been associated with protein degradation, DNA repair, cell 
cycle regulation, kinase modification, endocytosis, and regulation of other cell 
signaling pathways.

If we look for the highest dependency for the SW1088_CENTRAL_NERVOUS_SYSTEM cell
line, we see indeed that UBC (7316) is the most dependent gene for vitality. 

If perhaps we would like to look at the some of the lowest dependency scores:

```{r, eval=FALSE, echo=T}
# sort by 10 highest dependencies 
low10 <-df3[order(df3$Dependency_Score, decreasing=T)[1:10],]
low10
```

```{r, eval=FALSE, echo=T}
#sort dependencies scores of particular cell line my lowest (most important)
ubc <-rnai[order(rnai$SW1088_CENTRAL_NERVOUS_SYSTEM, decreasing=F)[1:10],]
dep_values_cns <- rbind(ubc$X1, ubc$SW1088_CENTRAL_NERVOUS_SYSTEM)
print("Dependency scores for top 10 dependent genes for SW1088_CENTRAL_NERVOUS_SYSTEM")
dep_values_cns
# [1] "Dependency scores for top 10 dependent genes for SW1088_CENTRAL_NERVOUS_SYSTEM"
#     [,1]             [,2]             [,3]             [,4]             [,5]             [,6]            
# [1,] "UBC (7316)"     "PSMA3 (5684)"   "PSMA4 (5685)"   "UBA52 (7311)"   "COPZ1 (22818)"  "SRSF1 (6426)"  
# [2,] "-5.93237872763" "-3.17473770863" "-2.95412602267" "-2.86619955206" "-2.68313273074" "-2.63257840872"
#     [,7]             [,8]            [,9]             [,10]                           
# [1,] "NACA (4666)"    "MED28 (80306)" "UBB (7314)"     "U2AF1L5&U2AF1 (102724594&7307)"
# [2,] "-2.52727165132" "-2.4743706891" "-2.31711569988" "-2.30982469365"        
```

