<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Theo Killian" />

<meta name="date" content="2019-02-19" />

<title>depmap vignette</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' || rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">depmap vignette</h1>
<h4 class="author">Theo Killian</h4>
<h4 class="date">February 19, 2019</h4>



<div id="depmap" class="section level1">
<h1><a href="https://github.com/tfkillian/depmap/">depmap</a></h1>
<p>This vignette illustrates the steps in loading of the data found in the <a href="https://github.com/tfkillian/depmap/">depmap</a> package.</p>
<div id="sec:intro" class="section level2">
<h2>Introduction</h2>
<p>The <a href="https://github.com/tfkillian/depmap/">depmap</a> package aims to provide a reproducible research framework to cancer dependency data described by <a href="https://www.ncbi.nlm.nih.gov/pubmed/28753430">Tsherniak, Aviad, et al. “Defining a cancer dependency map.” Cell 170.3 (2017): 564-576.</a>. The data found in the <a href="https://github.com/tfkillian/depmap/">depmap</a> package has been formatted to facilitate the use of common R packages such as <code>dplyr</code> and <code>ggplot2</code>. We hope that this package will allow researchers to more easily mine, explore and visually illustrate dependency data taken from the Depmap cancer genomic dependency study.</p>
</div>
<div id="preliminaries" class="section level2">
<h2>Preliminaries</h2>
<p>To install <a href="https://github.com/tfkillian/depmap/">depmap</a>, the <a href="https://cran.r-project.org/web/packages/BiocManager/index.html">BiocManager</a> Bioconductor Project Package Manager is required. If <a href="https://cran.r-project.org/web/packages/BiocManager/index.html">BiocManager</a> is not already installed, it will need to be done so beforehand. Type (within R) install.packages(“BiocManager”) (This needs to be done just once.)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)
BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&quot;UCLouvain-CBIO/depmap&quot;</span>)</code></pre></div>
</div>
<div id="sec:data" class="section level2">
<h2>Data Import</h2>
<p>The <a href="https://github.com/tfkillian/depmap/">depmap</a> package currently contains seven datasets in <em>.rda</em> format. These datasets can be loaded by using <em>data()</em> after this package has been installed and loaded in RStudio with <em>library(“depmap”)</em>. The data found in this R package has been converted from wide format to long format, although none of the values taken from the original datasets has been changed. The code related to the data transformation is can be found in the make_data.R file found in ./depmap/inst/scripts</p>
<div id="rnai" class="section level3">
<h3><code>rnai</code></h3>
<p>The <code>rnai</code> dataset contains the combined genetic dependency data for RNAi - induced gene knockdown for select genes and cancer cell lines. This data corresponds to the <code>D2_combined_genetic_dependency_scores.csv</code> file found in the most recent 19Q1 depmap release and includes 17309 genes, 711 cell lines 711, 30 primary diseases and 31 lineages.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&quot;rnai&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;depmap&quot;</span>)
<span class="kw">head</span>(rnai)
<span class="co">#&gt; # A tibble: 6 x 6</span>
<span class="co">#&gt;   depmap_ID  cell_Line          gene         gene_Name entrez_ID dependency</span>
<span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;              &lt;chr&gt;        &lt;chr&gt;     &lt;chr&gt;          &lt;dbl&gt;</span>
<span class="co">#&gt; 1 ACH-001270 127399_SOFT_TISSUE A1BG (1)     A1BG      1             NA    </span>
<span class="co">#&gt; 2 ACH-001270 127399_SOFT_TISSUE NAT2 (10)    NAT2      10            NA    </span>
<span class="co">#&gt; 3 ACH-001270 127399_SOFT_TISSUE ADA (100)    ADA       100           NA    </span>
<span class="co">#&gt; 4 ACH-001270 127399_SOFT_TISSUE CDH2 (1000)  CDH2      1000          -0.195</span>
<span class="co">#&gt; 5 ACH-001270 127399_SOFT_TISSUE AKT3 (10000) AKT3      10000         -0.256</span>
<span class="co">#&gt; 6 ACH-001270 127399_SOFT_TISSUE MED6 (10001) MED6      10001         -0.174</span></code></pre></div>
</div>
<div id="crispr" class="section level3">
<h3><code>crispr</code></h3>
<p>The <code>crispr</code> dataset contains the (batch corrected CERES inferred gene effect) CRISPR-Cas9 knockout data of select genes and cancer cell lines. This data corresponds to the <code>gene_effect_corrected.csv</code> file from the most recent 19Q1 depmap release. Data from this dataset includes 17634 genes, 558 cell lines, 26 primary diseases, 28 lineages.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&quot;crispr&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;depmap&quot;</span>)
<span class="kw">head</span>(crispr)
<span class="co">#&gt; # A tibble: 6 x 6</span>
<span class="co">#&gt;   depmap_ID  cell_Line                gene   gene_Name entrez_ID dependency</span>
<span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;                    &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;          &lt;dbl&gt;</span>
<span class="co">#&gt; 1 ACH-000004 HEL_HAEMATOPOIETIC_AND_… A1BG … A1BG      1             0.135 </span>
<span class="co">#&gt; 2 ACH-000005 HEL9217_HAEMATOPOIETIC_… A1BG … A1BG      1            -0.212 </span>
<span class="co">#&gt; 3 ACH-000007 LS513_LARGE_INTESTINE    A1BG … A1BG      1             0.0433</span>
<span class="co">#&gt; 4 ACH-000009 C2BBE1_LARGE_INTESTINE   A1BG … A1BG      1             0.0705</span>
<span class="co">#&gt; 5 ACH-000011 253J_URINARY_TRACT       A1BG … A1BG      1             0.191 </span>
<span class="co">#&gt; 6 ACH-000012 HCC827_LUNG              A1BG … A1BG      1            -0.0104</span></code></pre></div>
</div>
<div id="copynumber" class="section level3">
<h3><code>copyNumber</code></h3>
<p>The <code>copyNumber</code> dataset contains the WES copy number data, relating to the numerical log-fold copy number change measured against the baseline copy number of select genes and cell lines. This dataset corresponds to the <code>public_19Q1_gene_cn.csv</code> from the most recent 19Q1 depmap release. This dataset includes 23299 genes, 1604 cell lines, 38 primary diseases and 33 lineages.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&quot;copyNumber&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;depmap&quot;</span>)
<span class="kw">head</span>(copyNumber)
<span class="co">#&gt; # A tibble: 6 x 6</span>
<span class="co">#&gt;   depmap_ID  cell_Line             gene   gene_Name entrez_ID logCopyNumber</span>
<span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;                 &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;             &lt;dbl&gt;</span>
<span class="co">#&gt; 1 ACH-000011 253J_URINARY_TRACT    A1BG … A1BG      1               0.131  </span>
<span class="co">#&gt; 2 ACH-000026 253JBV_URINARY_TRACT  A1BG … A1BG      1              -0.237  </span>
<span class="co">#&gt; 3 ACH-000086 ACCMESO1_PLEURA       A1BG … A1BG      1               0.134  </span>
<span class="co">#&gt; 4 ACH-000557 AML193_HAEMATOPOIETI… A1BG … A1BG      1              -0.0208 </span>
<span class="co">#&gt; 5 ACH-000838 AMO1_HAEMATOPOIETIC_… A1BG … A1BG      1               0.170  </span>
<span class="co">#&gt; 6 ACH-000080 BDCM_HAEMATOPOIETIC_… A1BG … A1BG      1               0.00703</span></code></pre></div>
</div>
<div id="rppa" class="section level3">
<h3><code>RPPA</code></h3>
<p>The <code>RPPA</code> dataset contains the CCLE Reverse Phase Protein Array (RPPA) data. data which corresponds to the <code>CCLE_RPPA_20180123.csv</code> file from the most recent 19Q1 depmap release. This dataset includes 214 genes, 899 cell lines, 28 primary Diseases, 28 lineages.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&quot;RPPA&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;depmap&quot;</span>)
<span class="kw">head</span>(RPPA)
<span class="co">#&gt; # A tibble: 6 x 4</span>
<span class="co">#&gt;   depmap_ID  cell_Line                                antibody   expression</span>
<span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;                                    &lt;chr&gt;           &lt;dbl&gt;</span>
<span class="co">#&gt; 1 ACH-000698 DMS53_LUNG                               14-3-3_be…    -0.105 </span>
<span class="co">#&gt; 2 ACH-000489 SW1116_LARGE_INTESTINE                   14-3-3_be…     0.359 </span>
<span class="co">#&gt; 3 ACH-000431 NCIH1694_LUNG                            14-3-3_be…     0.0287</span>
<span class="co">#&gt; 4 ACH-000707 P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE 14-3-3_be…     0.120 </span>
<span class="co">#&gt; 5 ACH-000509 HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE 14-3-3_be…    -0.269 </span>
<span class="co">#&gt; 6 ACH-000522 UMUC3_URINARY_TRACT                      14-3-3_be…    -0.171</span></code></pre></div>
</div>
<div id="tpm" class="section level3">
<h3><code>TPM</code></h3>
<p>The <code>TPM</code> dataset contains the CCLE RNAseq gene expression data. This shows expression data only for protein coding genes (using scale log2(TPM+1)). This data corresponds to the <code>CCLE_depMap_19Q1_TPM.csv</code> file from the most recent 19Q1 depmap release. This dataset includes 57820 genes, 1165 cell lines, 33 primary Diseases, 32 lineages.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&quot;TPM&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;depmap&quot;</span>)
<span class="kw">head</span>(TPM)
<span class="co">#&gt; # A tibble: 6 x 6</span>
<span class="co">#&gt;   depmap_ID  cell_Line         gene        gene_Name ensembl_ID  expression</span>
<span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;             &lt;chr&gt;       &lt;chr&gt;     &lt;chr&gt;            &lt;dbl&gt;</span>
<span class="co">#&gt; 1 ACH-000956 22RV1_PROSTATE    TSPAN6 (EN… TSPAN6    ENSG000000…       2.65</span>
<span class="co">#&gt; 2 ACH-000948 2313287_STOMACH   TSPAN6 (EN… TSPAN6    ENSG000000…       3.00</span>
<span class="co">#&gt; 3 ACH-000026 253JBV_URINARY_T… TSPAN6 (EN… TSPAN6    ENSG000000…       4.57</span>
<span class="co">#&gt; 4 ACH-000011 253J_URINARY_TRA… TSPAN6 (EN… TSPAN6    ENSG000000…       4.58</span>
<span class="co">#&gt; 5 ACH-000323 42MGBA_CENTRAL_N… TSPAN6 (EN… TSPAN6    ENSG000000…       4.59</span>
<span class="co">#&gt; 6 ACH-000905 5637_URINARY_TRA… TSPAN6 (EN… TSPAN6    ENSG000000…       5.88</span></code></pre></div>
</div>
<div id="metadata" class="section level3">
<h3><code>metadata</code></h3>
<p>The <code>metadata</code> dataset contains the metadata about all of the cancer cell lines. It corresponds to the <code>depmap_19Q1_cell_lines.csv</code> file found in the most recent 19Q1 depmap release. This dataset includes 0 genes, 1677 cell lines, 38 primary diseases and 33 lineages.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&quot;metadata&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;depmap&quot;</span>)
<span class="kw">head</span>(metadata)
<span class="co">#&gt; # A tibble: 6 x 9</span>
<span class="co">#&gt;   depmap_ID cell_Line aliases COSMIC_ID sanger_ID primary_Disease</span>
<span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;       &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;          </span>
<span class="co">#&gt; 1 ACH-0000… NIHOVCAR… NIH:OV…    905933      2201 Ovarian Cancer </span>
<span class="co">#&gt; 2 ACH-0000… HL60_HAE… HL-60      905938        55 Leukemia       </span>
<span class="co">#&gt; 3 ACH-0000… CACO2_LA… CACO2;…        NA        NA Colon/Colorect…</span>
<span class="co">#&gt; 4 ACH-0000… HEL_HAEM… HEL        907053       783 Leukemia       </span>
<span class="co">#&gt; 5 ACH-0000… HEL9217_… HEL 92…        NA        NA Leukemia       </span>
<span class="co">#&gt; 6 ACH-0000… MONOMAC6… MONO-M…    908148      2167 Leukemia       </span>
<span class="co">#&gt; # … with 3 more variables: subtype_Disease &lt;chr&gt;, gender &lt;chr&gt;,</span>
<span class="co">#&gt; #   source &lt;chr&gt;</span></code></pre></div>
</div>
<div id="mutationcalls" class="section level3">
<h3><code>mutationCalls</code></h3>
<p>The <code>mutationCalls</code> dataset contains all merged mutation calls (coding region, germline filtered) found in the depmap dependency study. This dataset corresponds with the <code>depmap_19Q1_mutation_calls.csv</code> file found in the most recent 19Q1 depmap release and includes 18755 genes, 1601 cell lines, 37 primary diseases and 33 lineages.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&quot;mutationCalls&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;depmap&quot;</span>)
<span class="kw">head</span>(mutationCalls)
<span class="co">#&gt; # A tibble: 6 x 35</span>
<span class="co">#&gt;   depmap_ID Hugo_Symbol Entrez_Gene_Id NCBI_Build Chromosome Start_position</span>
<span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;                &lt;dbl&gt;      &lt;dbl&gt; &lt;chr&gt;               &lt;dbl&gt;</span>
<span class="co">#&gt; 1 ACH-0000… VPS13D               55187         37 1                12359347</span>
<span class="co">#&gt; 2 ACH-0000… AADACL4             343066         37 1                12726308</span>
<span class="co">#&gt; 3 ACH-0000… IFNLR1              163702         37 1                24484172</span>
<span class="co">#&gt; 4 ACH-0000… TMEM57               55219         37 1                25785018</span>
<span class="co">#&gt; 5 ACH-0000… ZSCAN20               7579         37 1                33954141</span>
<span class="co">#&gt; 6 ACH-0000… POU3F1                5453         37 1                38512139</span>
<span class="co">#&gt; # … with 29 more variables: End_position &lt;dbl&gt;, Strand &lt;chr&gt;,</span>
<span class="co">#&gt; #   Variant_Classification &lt;chr&gt;, Variant_Type &lt;chr&gt;,</span>
<span class="co">#&gt; #   Reference_Allele &lt;chr&gt;, Tumor_Seq_Allele1 &lt;chr&gt;, dbSNP_RS &lt;chr&gt;,</span>
<span class="co">#&gt; #   dbSNP_Val_Status &lt;chr&gt;, Genome_Change &lt;chr&gt;,</span>
<span class="co">#&gt; #   Annotation_Transcript &lt;chr&gt;, Tumor_Sample_Barcode &lt;chr&gt;,</span>
<span class="co">#&gt; #   cDNA_Change &lt;chr&gt;, Codon_Change &lt;chr&gt;, Protein_Change &lt;chr&gt;,</span>
<span class="co">#&gt; #   isDeleterious &lt;lgl&gt;, isTCGAhotspot &lt;lgl&gt;, TCGAhsCnt &lt;dbl&gt;,</span>
<span class="co">#&gt; #   isCOSMIChotspot &lt;lgl&gt;, COSMIChsCnt &lt;dbl&gt;, ExAC_AF &lt;dbl&gt;,</span>
<span class="co">#&gt; #   VA_WES_AC &lt;chr&gt;, CGA_WES_AC &lt;chr&gt;, SangerWES_AC &lt;chr&gt;,</span>
<span class="co">#&gt; #   SangerRecalibWES_AC &lt;chr&gt;, RNAseq_AC &lt;chr&gt;, HC_AC &lt;chr&gt;, RD_AC &lt;chr&gt;,</span>
<span class="co">#&gt; #   WGS_AC &lt;chr&gt;, Variant_annotation &lt;chr&gt;</span></code></pre></div>
</div>
<div id="downloading-broad-institute-data" class="section level3">
<h3>Downloading <a href="https://depmap.org/portal/download/">Broad Institute</a> data</h3>
<p>If desired, the original wide form data from which the long form datasets found in the <a href="https://github.com/tfkillian/depmap/">depmap</a> package were derived can be downloaded from the <a href="https://depmap.org/portal/download/">Broad Institute</a> website. The instructions on how to download these files and how the data was transformed and loaded into the <a href="https://github.com/tfkillian/depmap/">depmap</a> package can be found in the make_data.R file found in ./inst/scripts. (It should be noted that the original uncompressed <em>.csv</em> files are 1.5GB in total and take a moderate amount of time to download.)</p>
</div>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
