---
title: "depmap vignette"
author: "Theo Killian"
date: "February 19, 2019"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{depmap vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\usepackage[utf8]{inputenc}
---

```{r, include = FALSE, echo = FALSE, results = 'asis'}
knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
```

# [depmap](https://github.com/tfkillian/depmap/)

This vignette illustrates the steps in loading and visualizing the data found in
the [depmap](https://github.com/tfkillian/depmap/) package.  

## Introduction {#sec:intro}

The [depmap](https://github.com/tfkillian/depmap/) package aims to provide a 
reproducible research framework to cancer dependency data described by 
[Tsherniak, Aviad, et al. "Defining a cancer dependency map." Cell 170.3 (2017):
564-576.](https://www.ncbi.nlm.nih.gov/pubmed/28753430). The data found in the
[depmap](https://github.com/tfkillian/depmap/) package has been formatted to 
facilitate the use of common visualization packages such as `ggplot2`. We hope 
that this package will allow researchers to more easily, mine explore and 
visually illustrate dependency data taken from the Depmap cancer genomic 
dependency study.

```{r, eval=FALSE, echo=FALSE, warning=FALSE, include=FALSE}
# This code chunk makes sure that all the libraries used here are installed
packages <- c("knitr", "tidyr", "dplyr", "ggplot2", "plotly", "viridis", 
              "tidyverse", "grid", "gridExtra", "reshape2")
if ( length(missing_pkgs <- setdiff(packages, 
                                    rownames(installed.packages()))) > 0) {
  message("Installing missing package(s): ", 
          paste(missing_pkgs, collapse = ", "))
  install.packages(missing_pkgs)
}
```

## Preliminaries

To install [depmap](https://github.com/tfkillian/depmap/), the [BiocManager](https://cran.r-project.org/web/packages/BiocManager/index.html) 
Bioconductor Project Package Manager is required. If 
[BiocManager](https://cran.r-project.org/web/packages/BiocManager/index.html) is
not already installed, it will need to be done so beforehand. Type (within R)
install.packages("BiocManager") (This needs to be done just once.)

```{r, eval=FALSE, echo=TRUE}
install.packages("BiocManager")
BiocManager::install("tfkillian/depmap")
```

## Data Import {#sec:data}

The [depmap](https://github.com/tfkillian/depmap/) package currently contains 
seven datasets in *.rda* format. These datasets can be loaded by using *data()*
after this package has been installed and loaded in RStudio with
*library("depmap")*. The data found in this R package has been converted from 
wide format to long format, although none of the values taken from the original 
datasets have been changed. The code related to the data transformation is 
can be found in the make_data.R file found in ./depmap/inst/scripts 

```{r, eval=FALSE, echo=TRUE}
library("depmap")
#load(package_version())
```

### `rnai` 
The `rnai` dataset contains the combined genetic dependency data for RNAi-
induced gene knockdown for select genes and cancer cell lines. This data 
corresponds to the `D2_combined_genetic_dependency_scores.csv` file found in the
most recent 19Q1 depmap release and includes 17309 genes, 711 cell lines 711, 30
primary diseases and 31 lineages. 

```{r, eval=FALSE, echo=TRUE}
data("rnai", package = "depmap")
rnai
dim(rnai)
# [1] 12324008        6
head(rnai)
# # A tibble: 6 x 6
#   depmap_ID  cell_Line          gene         gene_Name entrez_ID dependency
#   <chr>      <chr>              <chr>        <chr>     <chr>          <dbl>
# 1 ACH-001270 127399_SOFT_TISSUE A1BG (1)     A1BG      1             NA    
# 2 ACH-001270 127399_SOFT_TISSUE NAT2 (10)    NAT2      10            NA    
# 3 ACH-001270 127399_SOFT_TISSUE ADA (100)    ADA       100           NA    
# 4 ACH-001270 127399_SOFT_TISSUE CDH2 (1000)  CDH2      1000          -0.195
# 5 ACH-001270 127399_SOFT_TISSUE AKT3 (10000) AKT3      10000         -0.256
# 6 ACH-001270 127399_SOFT_TISSUE MED6 (10001) MED6      10001         -0.174
```

### `crispr`
The `crispr` dataset contains the (batch corrected CERES inferred gene effect) 
CRISPR-Cas9 knockout data of select genes and cancer cell lines. This data 
corresponds to the `gene_effect_corrected.csv` file from the most recent 19Q1 
depmap release. Data from this dataset includes 17634 genes, 558 cell lines, 26 
primary diseases, 28 lineages. 

```{r, eval=FALSE, echo=TRUE}
data("crispr", package = "depmap")
crispr
dim(crispr)
# [1] 9839772       6
head(crispr)
# # A tibble: 6 x 6
#   depmap_ID  cell_Line                gene   gene_Name entrez_ID dependency
#   <chr>      <chr>                    <chr>  <chr>     <chr>          <dbl>
# 1 ACH-000004 HEL_HAEMATOPOIETIC_AND_… A1BG … A1BG      1             0.135 
# 2 ACH-000005 HEL9217_HAEMATOPOIETIC_… A1BG … A1BG      1            -0.212 
# 3 ACH-000007 LS513_LARGE_INTESTINE    A1BG … A1BG      1             0.0433
# 4 ACH-000009 C2BBE1_LARGE_INTESTINE   A1BG … A1BG      1             0.0705
# 5 ACH-000011 253J_URINARY_TRACT       A1BG … A1BG      1             0.191 
# 6 ACH-000012 HCC827_LUNG              A1BG … A1BG      1            -0.0104       
```

### `copyNumber` 
The `copyNumber` dataset contains the WES copy number data, relating to the 
numerical log-fold copy number change measured against copy number baseline of 
select genes and cell lines found in the depmap dependency study. This dataset 
corresponds to the `public_19Q1_gene_cn.csv` from the most recent 19Q1 depmap 
release. This dataset includes 23299 genes, 1604 cell lines, 38 primary diseases
and 33 lineages. 

```{r, eval=FALSE, echo=TRUE}
data("copyNumber", package = "depmap")
copyNumber
dim(copyNumber)
# [1] 37371596        6
head(copyNumber)
# # A tibble: 6 x 6
#   depmap_ID  cell_Line             gene   gene_Name entrez_ID logCopyNumber
#   <chr>      <chr>                 <chr>  <chr>     <chr>             <dbl>
# 1 ACH-000011 253J_URINARY_TRACT    A1BG … A1BG      1               0.131  
# 2 ACH-000026 253JBV_URINARY_TRACT  A1BG … A1BG      1              -0.237  
# 3 ACH-000086 ACCMESO1_PLEURA       A1BG … A1BG      1               0.134  
# 4 ACH-000557 AML193_HAEMATOPOIETI… A1BG … A1BG      1              -0.0208 
# 5 ACH-000838 AMO1_HAEMATOPOIETIC_… A1BG … A1BG      1               0.170  
# 6 ACH-000080 BDCM_HAEMATOPOIETIC_… A1BG … A1BG      1               0.00703
```

### `RPPA`
The `RPPA` dataset contains the CCLE Reverse Phase Protein Array (RPPA) data.
data which corresponds to the `CCLE_RPPA_20180123.csv` file from the most 
recent 19Q1 depmap release. This dataset includes 214 genes, 899 cell lines, 28 
primary Diseases, 28 lineages. 

```{r, eval=FALSE, echo=TRUE}
data("RPPA", package = "depmap")
RPPA
dim(RPPA)
# [1] 192386      4
head(RPPA)
# # A tibble: 6 x 4
#   depmap_ID  cell_Line                                antibody   expression
#   <chr>      <chr>                                    <chr>           <dbl>
# 1 ACH-000698 DMS53_LUNG                               14-3-3_be…    -0.105 
# 2 ACH-000489 SW1116_LARGE_INTESTINE                   14-3-3_be…     0.359 
# 3 ACH-000431 NCIH1694_LUNG                            14-3-3_be…     0.0287
# 4 ACH-000707 P3HR1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE 14-3-3_be…     0.120 
# 5 ACH-000509 HUT78_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE 14-3-3_be…    -0.269 
# 6 ACH-000522 UMUC3_URINARY_TRACT                      14-3-3_be…    -0.171 
```

### `TPM`
The `TPM` dataset contains the CCLE RNAseq gene expression data (using scale 
log2(TPM+1)). This shows expression data only for protein coding genes. This 
data corresponds to the `CCLE_depMap_19Q1_TPM.csv` file from the most recent 
19Q1 depmap release. This dataset includes 57820 genes, 1165 cell lines, 33 
primary Diseases, 32 lineages.

```{r, eval=FALSE, echo=TRUE}
data("TPM", package = "depmap")
TPM
dim(TPM)
# [1] 67360300        6
head(TPM)
# # A tibble: 6 x 6
#   depmap_ID  cell_Line         gene        gene_Name ensembl_ID  expression
#   <chr>      <chr>             <chr>       <chr>     <chr>            <dbl>
# 1 ACH-000956 22RV1_PROSTATE    TSPAN6 (EN… TSPAN6    ENSG000000…       2.65
# 2 ACH-000948 2313287_STOMACH   TSPAN6 (EN… TSPAN6    ENSG000000…       3.00
# 3 ACH-000026 253JBV_URINARY_T… TSPAN6 (EN… TSPAN6    ENSG000000…       4.57
# 4 ACH-000011 253J_URINARY_TRA… TSPAN6 (EN… TSPAN6    ENSG000000…       4.58
# 5 ACH-000323 42MGBA_CENTRAL_N… TSPAN6 (EN… TSPAN6    ENSG000000…       4.59
# 6 ACH-000905 5637_URINARY_TRA… TSPAN6 (EN… TSPAN6    ENSG000000…       5.88
```

### `metadata` 
The `metadata` dataset contains the metadata about all of the cancer cell lines.
It corresponds to the `depmap_19Q1_cell_lines.csv` file found in the most recent
19Q1 depmap release. This dataset includes 0 genes, 1677 cell lines, 38 primary
diseases and 33 lineages. 

```{r, eval=FALSE, echo=TRUE}
data("metadata", package = "depmap")
metadata
dim(metadata)
# [1] 1676    9
head(metadata)
# # A tibble: 6 x 9
#   depmap_ID cell_Line aliases COSMIC_ID sanger_ID primary_Disease
#   <chr>     <chr>     <chr>       <dbl>     <dbl> <chr>          
# 1 ACH-0000… NIHOVCAR… NIH:OV…    905933      2201 Ovarian Cancer 
# 2 ACH-0000… HL60_HAE… HL-60      905938        55 Leukemia       
# 3 ACH-0000… CACO2_LA… CACO2;…        NA        NA Colon/Colorect…
# 4 ACH-0000… HEL_HAEM… HEL        907053       783 Leukemia       
# 5 ACH-0000… HEL9217_… HEL 92…        NA        NA Leukemia       
# 6 ACH-0000… MONOMAC6… MONO-M…    908148      2167 Leukemia       
# # … with 3 more variables: subtype_Disease <chr>, gender <chr>,
# #   source <chr>
```

### `mutationCalls` 
The `mutationCalls` dataset contains all merged mutation calls (coding region, 
germline filtered) found in the depmap dependency study. This dataset 
corresponds with the `depmap_19Q1_mutation_calls.csv` file found in the most 
recent 19Q1 depmap release and includes 18755 genes, 1601 cell lines, 37 primary
diseases and 33 lineages. 

```{r, eval=FALSE, echo=TRUE}
data("mutationCalls", package = "depmap")
mutationCalls
dim(mutationCalls)
# [1] 1243145      35
head(mutationCalls)
# # A tibble: 6 x 35
#   depmap_ID Hugo_Symbol Entrez_Gene_Id NCBI_Build Chromosome Start_position
#   <chr>     <chr>                <dbl>      <dbl> <chr>               <dbl>
# 1 ACH-0000… VPS13D               55187         37 1                12359347
# 2 ACH-0000… AADACL4             343066         37 1                12726308
# 3 ACH-0000… IFNLR1              163702         37 1                24484172
# 4 ACH-0000… TMEM57               55219         37 1                25785018
# 5 ACH-0000… ZSCAN20               7579         37 1                33954141
# 6 ACH-0000… POU3F1                5453         37 1                38512139
# # … with 29 more variables: End_position <dbl>, Strand <chr>,
# #   Variant_Classification <chr>, Variant_Type <chr>,
# #   Reference_Allele <chr>, Tumor_Seq_Allele1 <chr>, dbSNP_RS <chr>,
# #   dbSNP_Val_Status <chr>, Genome_Change <chr>,
# #   Annotation_Transcript <chr>, Tumor_Sample_Barcode <chr>,
# #   cDNA_Change <chr>, Codon_Change <chr>, Protein_Change <chr>,
# #   isDeleterious <lgl>, isTCGAhotspot <lgl>, TCGAhsCnt <dbl>,
# #   isCOSMIChotspot <lgl>, COSMIChsCnt <dbl>, ExAC_AF <dbl>,
# #   VA_WES_AC <chr>, CGA_WES_AC <chr>, SangerWES_AC <chr>,
# #   SangerRecalibWES_AC <chr>, RNAseq_AC <chr>, HC_AC <chr>, RD_AC <chr>,
# #   WGS_AC <chr>, Variant_annotation <chr>
```

### Downloading [Broad Institute](https://depmap.org/portal/download/) data 

If desired, the original wide form data from which the long form datasets found 
in the [depmap](https://github.com/tfkillian/depmap/) package were derived can 
be downloaded from the [Broad Institute](https://depmap.org/portal/download/) 
website. The instructions on how to download these files and how the data was 
transformed and loaded into the [depmap](https://github.com/tfkillian/depmap/) 
package can be found in the make_data.R file found in ./inst/scripts. (It should
be noted that the original uncompressed *.csv* files are 1.5GB in total and 
take a moderate amount of time to download.)

## Visualisations

Perhaps the most interesting datasets found within the 
[depmap](https://github.com/tfkillian/depmap/) package are those that relate to 
the cancer gene dependency score, such as `rnai` and `crispr`. These datasets 
contain a score expressing how vital a particular gene is in terms of how lethal
the knockout/knockdown of that gene is on a target cell line. For example, a 
highly negative dependency score implyies that a cell line is highly dependent 
on that gene. 

We will demonstrate how to obtain individual dependency scores corresponding to 
a specific gene and cell lineage. For example, shown below is the dependency of 
a breast cancer lineage, such as `184A1_BREAST` has on a human tumor suppressor 
gene, like `BRCA1` when it is knocked down via RNAi. Shown below is the 
comparison for data found within the `rnai` dataset. This shows a score which is
slightly positive, indicating that the knockdown of this gene is slightly 
beneficial to the vitality of this cancer cell lineage. However, it may be 
insightful to put this single dependency score in context. 

```{r, eval=TRUE, echo=TRUE}
# load necessary libraries
library("dplyr")

#find dependency score for "BRCA1"" on "184A1_Breast""
dep_Score_BRCA1_184A1Breast <- rnai %>% 
    select(cell_Line, gene_Name, dependency) %>%
    filter(cell_Line =="184A1_BREAST", gene_Name == "BRCA1") 
dep_Score_BRCA1_184A1Breast
```

Shown below is the average dependency score for `BRCA1` for all cancer cell 
lines in the `rnai` dataset.

```{r, eval=TRUE, echo=TRUE}
#avg gene dependency for "BRCA1"
brca1_DS_Avg_RNAI <- rnai %>% select(gene_Name, dependency) %>%
    filter(gene_Name == "BRCA1") %>%
  summarise(mean_dependency_brca1 = mean(dependency, na.rm = TRUE))
brca1_DS_Avg_RNAI
```

Or to see the average gene dependency across all genes in the entire `rnai` 
dataset. As one can see below, the average dependency for an average gene in the
`rnai` dataset is slightly negative but close to zero.

```{r, eval=TRUE, echo=TRUE}
#avg gene dependency for all genes in `rnai` dataset
all_Gene_DS_Avg_RNAI <- rnai %>% select(gene_Name, dependency) %>%
  summarise(mean_dependency_all_genes_rnai = mean(dependency, na.rm = TRUE))
all_Gene_DS_Avg_RNAI
```

Sometimes it is difficult to find the subset with the exact gene name one wishes
to find. In this case, it is better to search by `entrez_ID`. For example, a 
[recent paper]{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6251792/} describes
gene knockdown of *NRF2* increases chemosensitivity in certain types of cancer. 
It might be interesting to see what interactions knockdown of this gene has on
other cancer cell lines. However, searching by *filter(gene_Name == "NRF2")* 
will not yield any results. We know from NCBI that the Entrez ID for this gene 
is "4780" and it is possible to search this dataset by that criteria. Here it 
can be shown that the gene name for *NRF2* in the `rnai` dataset is *NFE2L2*.

```{r, eval=TRUE, echo=TRUE}
#find cell lines with dependency corresponding to "entrez_ID == "4780"
nrf2 <- rnai %>% select(entrez_ID, cell_Line, gene_Name, dependency) %>%
    filter(entrez_ID == "4780") 
nrf2
```

Below the highest dependency scores via RNAI knock down of a specific gene, 
*NFE2L2* will be obtained and the cancer cell lines associated with those values
will be listed. It appears that the knockdown of this gene is strongly 
associated with cell death with in lung and kidney cancer cell lines.

```{r, eval=TRUE, echo=TRUE}
#find cell lines with dependency for "NFE2L2"
ds_NFE2L2_RNAI <- rnai %>% select(cell_Line, gene_Name, dependency) %>%
    filter(gene_Name == "NFE2L2") 

#sort dependencies scores of gene "NFE2L2" by lowest (most important)
low_NFE2L2_Vals <-ds_NFE2L2_RNAI[order(ds_NFE2L2_RNAI$dependency, 
                                       decreasing=FALSE)[1:10],]
low_NFE2L2_Vals
```

If we would like to obtain the top 10 lowest dependency scores for a 
particular cell line (for example `NCIH2066_LUNG`) along with the genes 
associated with those values:

```{r, eval=TRUE, echo=TRUE}
#find genes with dependency for cell_Line == "NCIH2066_LUNG"
ds_NCIH2066_LUNG_RNAI <- rnai %>% select(cell_Line, gene_Name, dependency) %>%
    filter(cell_Line == "NCIH2066_LUNG") 

#sort dep scores by lowest (most important)
low_NCIH2066_LUNG_Vals <-ds_NCIH2066_LUNG_RNAI[order(
    ds_NCIH2066_LUNG_RNAI$dependency, decreasing=FALSE)[1:10],]
low_NCIH2066_LUNG_Vals
```

Below shows the most significant genes that deplete cancer cell lines upon 
knockdown and their dependency scores for the entire `rnai` data.

```{r, eval=TRUE, echo=TRUE}
# select `rnai` columns cell_Line, gene_Name, dependency
ds_Gene_RNAI <- rnai %>% select(cell_Line, gene_Name, dependency)

#sort dep scores of by greatest dependency
low_10_Vals_RNAI <- ds_Gene_RNAI[order(ds_Gene_RNAI$dependency, 
                                         decreasing=FALSE)[1:10],]
low_10_Vals_RNAI
```

Below shows the least significant genes that induce cancer cell line vitality 
upon knockdown and their dependency scores for the entire `rnai` data. 
Unsurprisingly, we see high incidence of "TP53", a well known cancer driver.

```{r, eval=TRUE, echo=TRUE}
# select `rnai` columns cell_Line, gene_Name, dependency
ds_Gene_RNAI <- rnai %>% select(cell_Line, gene_Name, dependency)

#sort dep scores of by greatest dependency
high_10_Vals_RNAI <- ds_Gene_RNAI[order(ds_Gene_RNAI$dependency, 
                                         decreasing=TRUE)[1:10],]
high_10_Vals_RNAI
```

Below we will apply some of the same selections as shown in the above examples 
on the `crispr` gene knockout dataset and observe the difference between that 
dataset and `rnai`. First we will look at the most significant dependency scores
in the `crispr` dataset. As can be seen below, there is a different population
of significant genes with the highest dependency score.

```{r, eval=TRUE, echo=TRUE}
# select `crispr` columns cell_Line, gene_Name, dependency
ds_Gene_CRISPR <- crispr %>% select(cell_Line, gene_Name, dependency)

#sort dep scores of by greatest dependency
low_10_Vals_CRISPR <- ds_Gene_CRISPR[order(ds_Gene_CRISPR$dependency, 
                                         decreasing=FALSE)[1:10],]
low_10_Vals_CRISPR
```

First we will look at the least significant (most cancer inducing) dependency 
scores in the `crispr` dataset.

```{r, eval=TRUE, echo=FALSE}
# select `crispr` columns cell_Line, gene_Name, dependency
ds_Gene_CRISPR <- crispr %>% select(cell_Line, gene_Name, dependency)

#sort dep scores by most cancer inducing
high_10_Vals_CRISPR <- ds_Gene_CRISPR[order(ds_Gene_CRISPR$dependency, 
                                         decreasing=TRUE)[1:10],]
high_10_Vals_CRISPR
```

Here we will plot the difference in expression between the most signficant genes
found in the `crispr` and `rnai` datasets. 

```{r, eval=TRUE, echo=FALSE}
# load necessary libraries
library(reshape2)
library(plotly)
library(ggplot2)
library(gridExtra)
library(grid)
library(viridis)

# select `crispr` columns gene_Name, dependency
ds_Gene_CRISPR <- crispr %>% select(gene_Name, dependency)

#sort `crispr` dep scores by most cancer inducing
high_20_Vals_CRISPR <- ds_Gene_CRISPR[order(ds_Gene_CRISPR$dependency, 
                                         decreasing=TRUE)[1:20],]

#sort `crispr` dep scores by least cancer inducing
low_20_Vals_CRISPR <- ds_Gene_CRISPR[order(ds_Gene_CRISPR$dependency, 
                                         decreasing=FALSE)[1:20],]

# select `rnai` columns gene_Name, dependency
ds_Gene_RNAI <- rnai %>% select(gene_Name, dependency)

#sort `rnai` dep scores by most cancer inducing
high_20_Vals_RNAI <- ds_Gene_RNAI[order(ds_Gene_RNAI$dependency, 
                                         decreasing=TRUE)[1:20],]

#sort `rnai` dep scores by least cancer inducing
low_20_Vals_RNAI <- ds_Gene_RNAI[order(ds_Gene_RNAI$dependency, 
                                         decreasing=FALSE)[1:20],]

# rnai highest dep scores
p1 <- ggplot(low_20_Vals_RNAI) + aes(gene_Name, dependency, color = dependency) + 
    geom_point() +  scale_colour_viridis() + 
    theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
    ggtitle("RNAI")

# crispr highest dep scores 
p2 <- ggplot(low_20_Vals_CRISPR) + aes(gene_Name, dependency, color = dependency) + 
    geom_point() +  scale_colour_viridis() + 
    theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
    ggtitle("CRISPR ")

# rnai lowest dep scores
p3 <- ggplot(high_20_Vals_RNAI) + aes(gene_Name, dependency, color = dependency) + 
    geom_point() +  scale_colour_viridis() + 
    theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
    ggtitle("RNAI ")

# crispr lowest dep scores 
p4 <- ggplot(high_20_Vals_CRISPR) + aes(gene_Name, dependency, color = dependency) + 
    geom_point() +  scale_colour_viridis() + 
    theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
    ggtitle("CRISPR ")



p1 <-ggplot(high_20_Vals_CRISPR, aes(x=gene_Name, y=dependency))+
  geom_bar(stat='identity', fill="forest green")+
  ylab("dependency")+ theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  ggtitle("Lowest dep score genes CRISPR")
p1 


#plot as 1x2 grid
#grid.arrange(p1, p2, p3, p4, nrow = 2, top = "Comparison of CRISPR and RNAI")
low_Dep_Comp <- (p1 + p2)
low_Dep_Comp + facet_wrap(~dependency,  ncol=1)
```

Compare the top 100 unique genes for `crispr` and `rnai` 

```{r}
#compare count 

# select `rnai` columns cell_Line, gene_Name, dependency
ds_Gene_CRISPR <- crispr %>% select(cell_Line, gene_Name, dependency)

#sort dep scores by most cancer inducing
high_10_Vals_CRISPR <- ds_Gene_CRISPR[order(ds_Gene_CRISPR$dependency, 
                                         decreasing=TRUE)[1:10],]
high_10_Vals_CRISPR

```



```{r}


```



```{r, eval=FALSE, echo=T}
library(tidyverse, warn.conflicts = FALSE)
require(reshape2, warn.conflicts = FALSE)

## p5 - plot of dependency scores of 184A1_Breast
# omit genes with NA dependency and coerce into data frame
breastNoNA <-as.data.frame(na.omit(t(rnai0Chars[breastRow,])))

# add an ID column
breastNoNA <- tibble::rowid_to_column(breastNoNA, "ID")
Breast_184A1 <-breastNoNA

# add col names
colnames(Breast_184A1, do.NULL = FALSE)
colnames(Breast_184A1) <- c("Gene","Dependency")

# p1 - plot gene dependency levels for 184A1_BREAST cell line
p5 <-ggplot(Breast_184A1) + 
    aes(Gene, Dependency, color = Breast_184A1$Dependency) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=breastDSavg, linetype="dashed", color = "black") +
    ggtitle("Avg. RNAi dep. scores for each gene for cell line 184A1_Breast")

# p6 - boxplot of dependency scores of first 10 genes in `rnai`
# subset first 10 columns of `rnai` to create `first10Rnai`
first10Rnai <- rnai[3:13]

p6 <- ggplot(data = melt(first10Rnai), aes(x=variable, y=value)) + 
    geom_boxplot(aes(fill=variable)) + 
    theme(axis.text.x = element_text(angle = 90, hjust = 1)) + 
    ggtitle("Boxplot of dependency scores of first 10 genes in `rnai` dataset")

## p7 - plot of the lowest dep. scores for each gene for `rnai` data 
# apply function to get mean dependency of each gene from all columns
rnaiColAvg <- apply(rnai[c(3:17311)], 2, function(x) {mean(x, na.rm = TRUE)})
rnaiColAvgVal <- mean(rnaiColAvg)
# -0.06833473

#coerce into data frame
depAllGenes <- as.data.frame(t(rnaiColAvg))
depAllGenes <- tibble::rowid_to_column(depAllGenes, "ID")
names(depAllGenes)

#plot gene dependency levels for all cell lines
p7 <-ggplot(depAllGenes) + 
    aes(ID, Dependency_Score, color = depAllGenes$Dependency) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=rnaiColAvgVal, linetype="dashed", color = "black") + 
    geom_hline(yintercept=df3_avg, linetype="dashed", color = "red") +
    labs(y = "Dep Score of Gene with Largest Dep Score", x = "Cell Line") + 
    ggtitle("Greatest RNAi dependency scores for each gene by cell line") +
    scale_linetype_manual(name = "Avg dep scores",values = c(2,2))

# I will add a legend to the lines when I have time

#plot as 1x2 grid
grid.arrange(p5, p6, p7, nrow = 1, top = "Depmap Data Visualizations")

```

It might be interesting to locate the highest scoring dependencies among all 
cell lines for the CRISPR dataset as well.

```{r, eval=FALSE, echo=T}
# function that finds highest dependency scores for each gene and cell line

lowDepScoresCRISPR <- apply(crispr[c(2:17635)], 1, function(x) {min(x, na.rm = TRUE)})

#lowDepGeneCRISPR <- apply(crispr[c(2:17635)], 1, function(x) {depmin <-which.min(x) crispr[depmin]})

# coerce into data frame
lowDepScoresCRISPR <- as.data.frame(lowDepScoresCRISPR)
lowDepGeneCRISPR <- as.data.frame(lowDepGeneCRISPR)

df4<-cbind(lowDepGeneCRISPR, lowDepScoresCRISPR)

colnames(df4, do.NULL = FALSE)
colnames(df4) <- c("Gene","Dependency_Score", "Num")

# sort by 10 highest dependencies 
top10CRISPR <-df4[order(df4$Dependency_Score, decreasing=F)[1:10],]
top10CRISPR
```

Below is a plot of the lowest dependency scores for each gene and CL for the 
`crispr` dataset. 
 
```{r, eval=FALSE, echo=T}
# apply function to get mean dependency from all columns
gEc_avg <- apply(crispr[c(2:17635)], 1, function(x) {mean(x, na.rm = TRUE)})

gEc_avg2 <- mean(gEc_avg)
# -0.06833473

# apply function to get mean HIGHEST dependency from all columns
df4_avg <- mean(df4$Dependency_Score)
# -2.198807

#coerce into data frame
df4 <- as.data.frame(df4)
#df3 <- tibble::rowid_to_column(df3, "ID")
#View(df3)
depAllGenesCRISPR <-df4 

#plot gene dependency levels for all cell lines
ggplot(depAllGenesCRISPR) + aes(Num, Dependency_Score, color = depAllGenesCRISPR$Dependency_Score) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=gEc_avg2, linetype="dashed", color = "black") + 
    geom_hline(yintercept=df4_avg, linetype="dashed", color = "red") +
    labs(y = "Dep Score of Gene with Largest Dep Score", x = "Cell Line") + 
    ggtitle("Greatest CRISPR dependency scores for each gene by cell line") +
    scale_linetype_manual(name = "Avg dep scores",values = c(2,2))

# I will add a legend to the lines when I have time
```

It might be insightful to compare the dependency scores of datasets derived from
RNAi knockout e.g. `rnai` and those utilizing CRISPR knockouts e.g. `crispr`. 

```{r, eval=TRUE, echo=F}
# data wrangling for `crispr`

# Rename first column to "cellLine" to it can be merged with `metadata`
names(crispr)[1]<-"depmapID"

# map CCLE name onto DepmapID for crispr
geneToID <- full_join(crispr, metadata, all=TRUE)

# tidy up dataframe to make cellLine appear as the second column
depmapID <- geneToID$depmapID
cellLine <- geneToID$cellLine
geneToIDCols <- subset(cellToID3, select = -c(depmapID, cellLine))
geneIDcrispr <-cbind(depmapID, cellLine, geneToID)
geneIDcrispr <- geneIDcrispr[,-3] #remove reduncent depmapID column
```

```{r, eval=TRUE, echo=F}
# in order to plot the `crispr` (CRISPR) and `rnai`(RNAi) datasets, we will have
# to make them comparable, which means removing any NAs or genes that were not 
# tested in on or the other dataset. We will 1) remove all rows with NA for 
# dependency score for the PIK3C3 gene and 2) remove all genes (columns) that do
# not occur in both datasets, then 3) full_join on the cell lines (depmapID) in 
# common for the CRISPR and RNAi for the PIK3C3 gene dependencies of both 
# datasets 

# compare `depmapID` column of `geneIDcrispr` and `noNAdepIDrnai` 

# since the dimensions for `geneIDcrispr` and `noNAdepIDrnai` are the same, we
# will full_join on `depmapID` and create a new dataframe `depPIK3C3` with 
# depenency scores that can be compared between RNAi CRISPR datasets. Note: we
# use "cbind.data.frame" to avoid turning the numeric values into factors

crisprPIK3C3 <-cbind.data.frame(geneIDcrispr$depmapID, geneIDcrispr$`PIK3C3 (5289)`)
rnaiPIK3C3 <-cbind.data.frame(noNAdepIDrnai$depmapID, noNAdepIDrnai$`PIK3C3 (5289)`, stringsAsFactors = FALSE)

#rename column names 

#add col names to crisprPIK3C3 to make the variables more clear
colnames(crisprPIK3C3, do.NULL = FALSE)
colnames(crisprPIK3C3) <- c("depmapID","crisprPIK3C3")

#add col names to rnaiPIK3C3 to make the variables more clear
colnames(rnaiPIK3C3, do.NULL = FALSE)
colnames(rnaiPIK3C3) <- c("depmapID","rnaiPIK3C3")

#full_join on depmapID
comparePIK3C3 <- full_join(crisprPIK3C3, rnaiPIK3C3, all=TRUE)

#set all NAs to 0 for comparePIK3C3 so the dataframe can be visualized

# comparePIK3C3$rnaiPIK3C3 has to be turned into character because it throws 
# an annoying warning "invalid factor level, NA generated"
comparePIK3C3$rnaiPIK3C3<- as.character(comparePIK3C3$rnaiPIK3C3)
comparePIK3C3[, 1:3][is.na(comparePIK3C3[, 1:3])] <- 0

# comparePIK3C3$rnaiPIK3C3 converted back to numeric. Note: this is an inelegant
# solution, need to find something that works better 
comparePIK3C3$rnaiPIK3C3<- as.numeric(comparePIK3C3$rnaiPIK3C3)

#convert comparePIK3C3 to depPIK3C3, and convert 0 values to NA
depPIK3C3 <- comparePIK3C3
depPIK3C3[depPIK3C3==0] <- NA
depPIK3C3

#Delete the rows associated with NA.
compDepPIK3C3<-depPIK3C3[complete.cases(depPIK3C3),]
#View(compDepPIK3C3)
#dim(compDepPIK3C3)
#[1] 402   3
# this means that there are only 402 cell lines where there are dependency 
# scores for both RNAI and CRISPR. we will plot them in the chunk below

# sort byDepIdPIK3C3 by depmapID, lowest to highest
byDepIdPIK3C3 <- compDepPIK3C3[order(compDepPIK3C3$depmapID, decreasing=F),]
```

Now that the data wrangling for `rnai` and 
`crispr` is complete, we will attempt to visualize the difference
in dependency scores between the RNAi and CRISPR datasets. First, comparison for 
the dependency two genes between the two datasets will be produced, then a 
global comparison between the two datasets will be demonstrated.

We will begin by illustrating a histogram of the dependency scores for a 
particular gene, comparing CRISPR and RNAI.

Scatterplot of the dependency scores for RNAi knockouts for PIK3C3 using ggplot2

```{r, eval=FALSE, echo=T}
# histogram of the dependency scores for a particular gene taken from data-
# wrangled `crispr` and `rnai` datasets. We will first be visualizing the PIK3C3
# gene, which is one of # the most dependent genes in either dataset. We will do
# this by combining the geneIDcrispr$`PIK3C3 (5289)` and 
# noNAdepIDrnai$`PIK3C3 (5289)` columns into a single dataframe and plotting 
# with ggplot2 

avgCRISPR <-mean(byDepIdPIK3C3$crisprPIK3C3)
avgRNAI <-mean(byDepIdPIK3C3$rnaiPIK3C3)

#plot RNAI gene dependency levels for PIK3C3 for all complete cell lines 
ggplot(byDepIdPIK3C3) + aes(depmapID, byDepIdPIK3C3$rnaiPIK3C3, color = byDepIdPIK3C3$rnaiPIK3C3) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=avgRNAI, linetype="dashed", color = "red") +
    labs(y = "Dependency Score", x = "Cell Lines") + 
    ggtitle("Plot of the dependency scores for RNAi for PIK3C3") 
```

Scatterplot of dependency scores for CRISPR knockouts for PIK3C3 using ggplot2

```{r, eval=FALSE, echo=T}
#plot CRISPR gene dependency levels for PIK3C3 for all complete cell lines
ggplot(byDepIdPIK3C3) + aes(depmapID, byDepIdPIK3C3$crisprPIK3C3, color = byDepIdPIK3C3$crisprPIK3C3) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=avgCRISPR, linetype="dashed", color = "blue") + 
    labs(y = "Dependency Score", x = "Cell Lines") + 
    ggtitle("Plot of the dependency scores for CRISPR for PIK3C3") 
```

Overlay of dependency scores for RNAi and CRISPR knockouts for PIK3C3 

```{r, eval=FALSE, echo=T}
#overlay RNAI and CRISPR
library(lattice)
xyplot(crisprPIK3C3 + rnaiPIK3C3 ~ sequence(nrow(byDepIdPIK3C3)), data=byDepIdPIK3C3, 
       pch = 21, fill = c("blue", "red"), cex = 1, xlab=list("Cell Lines", cex=1.7), 
       ylab=list("Dependency Score for Gene", cex=1.5), 
       main=list("Dependency Score of PIKC3C for RNAi and CRISPR for all cell lines", cex=1.1),
       auto.key=list(space = "top", column = 2, title="Knockout Type", cex.title=1),
       panel = function(x, y, ...)
       {panel.xyplot(x, y, ...)  ## First call the default panel function for 'xyplot'
       panel.abline(h = avgCRISPR, col = 2, lty = 2)
       panel.abline(h = avgRNAI, col = 1, lty = 2)
       })
```

Plot CRISPR vs RNAi gene dependency levels for PIK3C3

```{r, eval=FALSE, echo=T}
#plot CRISPR vs RNAI gene dependency levels for PIK3C3 for all complete cell lines 
ggplot(byDepIdPIK3C3) + aes(rnaiPIK3C3, crisprPIK3C3, color = byDepIdPIK3C3$rnaiPIK3C3) + 
    geom_point() +  scale_colour_viridis() + 
    labs(y = "CRISPR Dependency Score", x = "RNAI Dependency Score") + 
    ggtitle("Dependency scores for RNAi vs CRISPR for PIKC3C") 
```

Compare CRISPR vs RNAi for COPB1

```{r, eval=FALSE, echo=T}
# since the dimensions for `geneIDcrispr` and `noNAdepIDrnai` are the same, we
# will full_join on `depmapID` and create a new dataframe `depCOPB1` with 
# depenency scores that can be compared between RNAi CRISPR datasets. Note: we
# use "cbind.data.frame" to avoid turning the numeric values into factors

crisprCOPB1 <-cbind.data.frame(geneIDcrispr$depmapID, geneIDcrispr$`COPB1 (1315)`)
rnaiCOPB1 <-cbind.data.frame(noNAdepIDrnai$depmapID, noNAdepIDrnai$`COPB1 (1315)`, stringsAsFactors = FALSE)

#rename column names 

#add col names to crisprCOPB1 to make the variables more clear
colnames(crisprCOPB1, do.NULL = FALSE)
colnames(crisprCOPB1) <- c("depmapID","crisprCOPB1")

#add col names to rnaiCOPB1 to make the variables more clear
colnames(rnaiCOPB1, do.NULL = FALSE)
colnames(rnaiCOPB1) <- c("depmapID","rnaiCOPB1")

#full_join on depmapID
compareCOPB1 <- full_join(crisprCOPB1, rnaiCOPB1, all=TRUE)
#View(compareCOPB1)

compDepCOPB1<-compareCOPB1[complete.cases(compareCOPB1),]
#View(compDepCOPB1)

compDepCOPB1$rnaiCOPB1<- as.numeric(as.character(compDepCOPB1$rnaiCOPB1))

byDepIdCOPB1 <- compDepCOPB1
```

CRISPR vs RNAi for COPB1

```{r, eval=FALSE, echo=T}
#plot CRISPR vs RNAI gene dependency levels for COPB1 for all complete cell lines 
ggplot(byDepIdCOPB1) + aes(rnaiCOPB1, crisprCOPB1, color = byDepIdCOPB1$rnaiCOPB1) + 
    geom_point() +  scale_colour_viridis() + 
    labs(y = "CRISPR Dependency Score", x = "RNAI Dependency Score") + 
    ggtitle("Dependency scores for RNAi vs CRISPR for COPB1") 
```


Overlay of dependency scores for RNAi and CRISPR knockouts for COPB1 

```{r, eval=FALSE, echo=T}
avgCRISPR2 <-mean(byDepIdPIK3C3$crisprCOPB1)
avgRNAI2 <-mean(byDepIdPIK3C3$rnaiCOPB1)

#overlay RNAI and CRISPR
library(lattice)
xyplot(crisprCOPB1 + rnaiCOPB1 ~ sequence(nrow(byDepIdCOPB1)), data=byDepIdCOPB1, 
       pch = 21, fill = c("blue", "red"), cex = 1, xlab=list("Cell Lines", cex=1.7), 
       ylab=list("Dependency Score for Gene", cex=1.5), 
       main=list("Dependency Score of COPB1 for RNAi and CRISPR for all cell lines", cex=1.1),
       auto.key=list(space = "top", column = 2, title="Knockout Type", cex.title=1),
       panel = function(x, y, ...)
       {panel.xyplot(x, y, ...)  ## First call the default panel function for 'xyplot'
       panel.abline(h = avgCRISPR2, col = 2, lty = 2)
       panel.abline(h = avgRNAI2, col = 1, lty = 2)
       })
```

Some data wrangling must be performed to visualize the differences in dependency 
for both RNAi and CRISPR datasets for all genes. 

```{r, eval=TRUE, echo=F}
# apply function to calculate the mean of all of the dependencies for each gene

# convert geneIDcrispr to geneIDcrispr2 so we can preserve original df
geneIDcrispr2 <- geneIDcrispr

# convert noNAdepIDrnai to noNAdepIDrnai2 so we can preserve original df
noNAdepIDrnai2 <- noNAdepIDrnai

#nremove columns with characters
geneIDcrispr2sub <- geneIDcrispr2[3:17311]

crisprCellMean <- apply(geneIDcrispr2sub, 1, mean, na.rm = TRUE)

crisprGeneMean <- apply(geneIDcrispr2sub, 2, mean, na.rm = TRUE)

#remove columns with characters
noNAdepIDrnai2sub <- noNAdepIDrnai2[3:17311]

# the columns of noNAdepIDrnai2sub were converted to factors somehow, and we
# will be converting them back to numeric in order to plot or manipulate them
indx <- sapply(noNAdepIDrnai2sub, is.factor)
noNAdepIDrnai2sub[indx] <- lapply(noNAdepIDrnai2sub[indx], 
                                  function(x) as.numeric(as.character(x)))

rnaiCellMean <- apply(noNAdepIDrnai2sub, 1, mean, na.rm = TRUE)

rnaiGeneMean <- apply(noNAdepIDrnai2sub, 2, mean, na.rm = TRUE)

# now we need to make combined dataframes of CellMean and GeneMean
# (it's easier to manipulate and plot if they are in one dataframe)

# cbind the CellMeans
cellMeans <-cbind(crisprCellMean, rnaiCellMean)

#add col names to cellMeans to make the variables more clear
colnames(cellMeans, do.NULL = FALSE)
colnames(cellMeans) <- c("CRISPRavgCellDep","RNAIavgCellDep")

#coerce into data frame
cellMeans <- as.data.frame(cellMeans)

# add sequence number 
cellMeans <- tibble::rowid_to_column(cellMeans, "ID")

# cbind the GeneMeans
geneMeans <-cbind(crisprGeneMean, rnaiGeneMean)

#add col names to cellMeans to make the variables more clear
colnames(geneMeans, do.NULL = FALSE)
colnames(geneMeans) <- c("CRISPRavgGeneDep","RNAIavgGeneDep")

#coerce into data frame
geneMeans <- as.data.frame(geneMeans)

# add sequence number 
geneMeans <- tibble::rowid_to_column(geneMeans, "ID")

# there are a lot of missing averages in CellMeans (oddly enough), we will only
# visualise the complete cases

#Delete the rows associated with NA.
compCellMeans<-cellMeans[complete.cases(cellMeans),]

#Delete the rows associated with NA.
compGeneMeans<-geneMeans[complete.cases(geneMeans),]

# calculate averages to show in plots

# total average CRISPR Gene dependency for all genes
totalAvgCrisprGene <- mean(geneMeans$CRISPRavgGeneDep, na.rm = TRUE)

# total average RNAi Gene dependency for all genes
totalAvgRNAiGene <- mean(geneMeans$RNAIavgGeneDep, na.rm = TRUE)

# total average CRISPR Gene dependency for all cell lines
totalAvgCrisprCell <- mean(cellMeans$CRISPRavgCellDep, na.rm = TRUE)

# total average RNAi Gene dependency for all cell lines
totalAvgRNAiCell <- mean(cellMeans$RNAIavgCellDep, na.rm = TRUE)
```

Plot of avg CRISPR dep scores for all Cell Lines

```{r, eval=FALSE, echo=T}
#Plot of avg CRISPR dep scores for all genes
ggplot(compCellMeans) + aes(ID, CRISPRavgCellDep, color = CRISPRavgCellDep) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=totalAvgCrisprCell, linetype="dashed", color = "blue") + 
    labs(y = "Average dependency score per cell line", x = "Cell Lines") + 
    ggtitle("Average CRISPR dependency scores for all Cell Lines") 
```

Plot of average RNAI dependency scores for all cell lines

```{r, eval=FALSE, echo=T}
#Plot of avg RNAI dep scores for all genes
ggplot(compCellMeans) + aes(ID, RNAIavgCellDep, color = RNAIavgCellDep) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=totalAvgRNAiCell, linetype="dashed", color = "blue") + 
    labs(y = "Average dependency score per cell line", x = "Cell Lines") + 
    ggtitle("Average RNAI dependency scores for all cell lines") 
```

Plot of average CRISPR dependency scores for all genes

```{r, eval=FALSE, echo=T}
#Plot of avg RNAI dep scores for all genes
ggplot(compGeneMeans) + aes(ID, CRISPRavgGeneDep, color = CRISPRavgGeneDep) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=totalAvgCrisprGene, linetype="dashed", color = "blue") + 
    labs(y = "Average dependency score per gene", x = "Genes") + 
    ggtitle("Average CRISPR dependency scores for all genes") 
```

Plot of avg RNAI dep scores for all genes

```{r, eval=FALSE, echo=T}
#Plot of avg RNAI dep scores for all cell lines
ggplot(compGeneMeans) + aes(ID, RNAIavgGeneDep, color = RNAIavgGeneDep) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=totalAvgRNAiGene, linetype="dashed", color = "blue") + 
    labs(y = "Dependency Score", x = "Genes") + 
    ggtitle("Average RNAi dependency scores for all genes") 
```


Plot CRISPR vs RNAi average gene dependency levels all cell lines

```{r, eval=FALSE, echo=T}
#plot CRISPR vs RNAi average gene dependency levels all genes
p2 <- ggplot(compGeneMeans) + aes(CRISPRavgGeneDep, RNAIavgGeneDep, color = CRISPRavgGeneDep) + 
    geom_point() +  scale_colour_viridis() + geom_smooth(method='lm',formula=y~x) +
    labs(y = "RNAIavgGeneDep", x = "CRISPRavgGeneDep") + 
    ggtitle("CRISPR vs RNAi average gene dependency levels all genes") 
p2 + geom_abline(intercept = 0, slope = 1)
```

Plot CRISPR vs RNAi average gene dependency levels all cells

```{r, eval=FALSE, echo=T}
#plot CRISPR vs RNAi average gene dependency levels all cells
p1 <- ggplot(compCellMeans) + aes(CRISPRavgCellDep, RNAIavgCellDep, color = CRISPRavgCellDep) + 
    geom_point() +  scale_colour_viridis() + geom_smooth(method='lm',formula=y~x) +
    labs(y = "RNAIavgCellDep", x = "CRISPRavgCellDep") + 
    ggtitle("CRISPR vs RNAi average gene dependency levels all cell lines") 
p1 + geom_abline(intercept = 0, slope = 1)
```

Average Dependency Scores for RNAi and CRISPR for all genes

```{r, eval=FALSE, echo=T}
#overlay RNAI and CRISPR dependency scores for genes
xyplot(CRISPRavgGeneDep + RNAIavgGeneDep ~ sequence(nrow(compGeneMeans)), data=compGeneMeans, 
       pch = 21, fill = c("blue", "red"), cex = 1, xlab=list("Genes", cex=1.7), 
       ylab=list("Avg Dep Score for each gene", cex=1.5), 
       main=list("Average Dependency Scores for RNAi and CRISPR for all genes", cex=1.1),
       auto.key=list(space = "top", column = 2, title="Knockout Type", cex.title=1),
       panel = function(x, y, ...)
       {panel.xyplot(x, y, ...)  ## First call the default panel function for 'xyplot'
       panel.abline(h = totalAvgCrisprGene, col = 2, lty = 2)
       panel.abline(h = totalAvgRNAiGene, col = 1, lty = 2)
       })
```

Average Dependency Scores for RNAi and CRISPR for all cell lines

```{r, eval=FALSE, echo=T}
#overlay RNAI and CRISPR for genes
xyplot(CRISPRavgCellDep + RNAIavgCellDep ~ sequence(nrow(compCellMeans)), data=compCellMeans, 
       pch = 21, fill = c("blue", "red"), cex = 1, xlab=list("Cell Lines", cex=1.7), 
       ylab=list("Avg Dep Score for each cell line", cex=1.5), 
       main=list("Average Dependency Scores for RNAi and CRISPR for all cell lines", cex=1.1),
       auto.key=list(space = "top", column = 2, title="Knockout Type", cex.title=1),
       panel = function(x, y, ...)
       {panel.xyplot(x, y, ...)  ## First call the default panel function for 'xyplot'
       panel.abline(h = totalAvgCrisprCell, col = 2, lty = 2)
       panel.abline(h = totalAvgRNAiCell, col = 1, lty = 2)
       })
```

(4) explore the dependency scores for genes with high and low expression 

PSMC2 (Rpt1) was the highest-ranked CYCLOPS candidate gene in the original 
depmap analysis and was also significant in the validation data set. PSMC2 is 
part of the 19S regulatory complex of the 26S proteasome, which is responsible 
for catalyzing the unfolding and translocation of substrates into the 20S 
proteasome

Copy Number Note: Find out what the values of copy number are in copyNumber

```{r, eval=TRUE, echo=F}

# apply function to calculate the mean of all of the dependencies for each gene

#nremove columns with characters
CNsub <- copyNumber[3:23300]

copyNumGeneMean <- apply(CNsub, 1, mean, na.rm = TRUE)

copyNumCellMean <- apply(CNsub, 2, mean, na.rm = TRUE)

#coerce into data frame
copyNumGeneMean <- as.data.frame(copyNumGeneMean)
copyNumCellMean <- as.data.frame(copyNumCellMean)

# add sequence number 
copyNumGeneMean <- tibble::rowid_to_column(copyNumGeneMean, "ID")
copyNumCellMean <- tibble::rowid_to_column(copyNumCellMean, "ID")

#add col names to cellMeans to make the variables more clear
colnames(copyNumCellMean, do.NULL = FALSE)
colnames(copyNumCellMean) <- c("ID","copyNumCellMean")

#add col names to cellMeans to make the variables more clear
colnames(copyNumGeneMean, do.NULL = FALSE)
colnames(copyNumGeneMean) <- c("ID","copyNumGeneMean")

#Delete the rows associated with NA.
copyNumGeneMean<-geneMeans[complete.cases(copyNumGeneMean),]
copyNumCellMean<-geneMeans[complete.cases(copyNumCellMean),]

# calculate averages to show in plots

# total average CRISPR Gene dependency for all genes
totalAvgCNGene <- mean(copyNumGeneMean$copyNumGeneMean, na.rm = TRUE)
#totalAvgCrisprGene 

# total average RNAi Gene dependency for all genes
totalAvgCNCell <- mean(copyNumCellMean$copyNumCellMean, na.rm = TRUE)
```

Plot of avg CRISPR dep scores for all Cell Lines

```{r, eval=FALSE, echo=T}
#Plot of avg CRISPR dep scores for all genes
ggplot(compCellMeans) + aes(ID, CRISPRavgCellDep, color = CRISPRavgCellDep) + 
    geom_point() +  scale_colour_viridis() + 
    geom_hline(yintercept=totalAvgCrisprCell, linetype="dashed", color = "blue")
    + labs(y = "Average dependency score per cell line", x = "Cell Lines") + 
    ggtitle("Average CRISPR dependency scores for all Cell Lines") 
```

Scatterplot overlay

```{r, eval=FALSE, echo=T}
#combine two scatterplots in one
ggplot(df) + geom_point(aes(x = x1, y = y1)) + geom_point(aes(x = x2, y = y2)
```

############## TO FIX ############################

Here we will locate the highest scoring gene dependencies among all cell lines 
for the RNAI dataset. These dependencies might make for interesting targets for 
research.

```{r, eval=FALSE, echo=T}
# remove character columns of `rnai` dataset rename to `rnai0Chars`
rnai0Chars <- subset(rnai, select = -c(depmapID, cellLine))

#add row names so that they can be retrieved later
row.names(rnai0Chars) <- rnai$cellLine

# apply function finds lowest value dep. scores for each gene (column)
lowDepScores <- apply(rnai0Chars, 1, function(x) {min(x, na.rm = TRUE)})

# apply function finds (row) of lowest value dep. scores for each gene (column)
lowDepCellLine <- apply(rnai[c(3:17311)], 1, function(x) which.min(x))
lowDepCellLine <- as.data.frame(lowDepCellLine, row.names = FALSE)
colnames(lowDepCellLine, do.NULL = FALSE)
colnames(lowDepCellLine) <- c("cellLine")
View(lowDepCellLine)

lowDepCellLineList <- rnai$depmapID[lowDepCellLine,]

View(lowDepCellLineList[c(1:10)])

# coerce into data frame
lowDepScores <- as.data.frame(lowDepScores)
View(lowDepScores)
#lowDepCellLine <- as.data.frame(lowDepCellLine)

# turn row names into a column
r1 <-rownames(lowDepScores)
lowDepScores$Cell_Line <- r1
r2 <-rownames(lowDepCellLine)
lowDepCellLine$Cell_Line <- r2

# tidy up df, merge on Cell_Line
df3 = merge(lowDepCellLine, lowDepScores, by.x=c("Cell_Line"), by.y=c("Cell_Line"))
colnames(df3, do.NULL = FALSE)
colnames(df3) <- c("Cell_Line", "Gene", "Dependency_Score")

# sort by 10 highest dependencies 
top10 <-df3[order(df3$Dependency_Score, decreasing=F)[1:10],]
top10
```

It might be interesting to locate the highest scoring dependencies among all 
cell lines for the CRISPR dataset as well.

```{r, eval=FALSE, echo=T}
# function that finds highest dependency scores for each gene and cell line

lowDepScoresCRISPR <- apply(crispr[c(2:17635)], 1, function(x) {min(x, na.rm = TRUE)})

lowDepGeneCRISPR <- apply(crispr[c(2:17635)], 1, function(x) {depmin <-which.min(x) crispr$depmapID[depmin]})

# coerce into data frame
lowDepScoresCRISPR <- as.data.frame(lowDepScoresCRISPR)
lowDepGeneCRISPR <- as.data.frame(lowDepGeneCRISPR)

df4<-cbind(lowDepGeneCRISPR, lowDepScoresCRISPR)

colnames(df4, do.NULL = FALSE)
colnames(df4) <- c("Gene","Dependency_Score", "Num")

# sort by 10 highest dependencies 
top10CRISPR <-df4[order(df4$Dependency_Score, decreasing=F)[1:10],]
top10CRISPR
```

The gene with the highest dependency score is UBC (7316). This gene represents a 
ubiquitin gene, ubiquitin C. The encoded protein is a polyubiquitin precursor.  
Ubiquitination has been associated with protein degradation, DNA repair, cell 
cycle regulation, kinase modification, endocytosis, and regulation of other cell 
signaling pathways.

If we look for the highest dependency for the SW1088_CENTRAL_NERVOUS_SYSTEM cell
line, we see indeed that UBC (7316) is the most dependent gene for vitality. 

If perhaps we would like to look at the some of the lowest dependency scores:

```{r, eval=FALSE, echo=T}
# sort by 10 highest dependencies 
low10 <-df3[order(df3$Dependency_Score, decreasing=T)[1:10],]
low10
```

```{r, eval=FALSE, echo=T}
#sort dependencies scores of particular cell line my lowest (most important)
ubc <-rnai[order(rnai$SW1088_CENTRAL_NERVOUS_SYSTEM, decreasing=F)[1:10],]
dep_values_cns <- rbind(ubc$X1, ubc$SW1088_CENTRAL_NERVOUS_SYSTEM)
print("Dependency scores for top 10 dependent genes for SW1088_CENTRAL_NERVOUS_SYSTEM")
dep_values_cns
# [1] "Dependency scores for top 10 dependent genes for SW1088_CENTRAL_NERVOUS_SYSTEM"
#     [,1]             [,2]             [,3]             [,4]             [,5]             [,6]            
# [1,] "UBC (7316)"     "PSMA3 (5684)"   "PSMA4 (5685)"   "UBA52 (7311)"   "COPZ1 (22818)"  "SRSF1 (6426)"  
# [2,] "-5.93237872763" "-3.17473770863" "-2.95412602267" "-2.86619955206" "-2.68313273074" "-2.63257840872"
#     [,7]             [,8]            [,9]             [,10]                           
# [1,] "NACA (4666)"    "MED28 (80306)" "UBB (7314)"     "U2AF1L5&U2AF1 (102724594&7307)"
# [2,] "-2.52727165132" "-2.4743706891" "-2.31711569988" "-2.30982469365"        
```


```{r, eval=TRUE, echo=FALSE, fig.width=10}
library(ggplot2, warn.conflicts = FALSE) 
library(viridis, warn.conflicts = FALSE)

#select data columns
cancerTypes <-unique(metadata$Primary.Disease)
numCancer <-(as.integer(cancerTypes))

#plot pie
pie(x=numCancer, label = "", init.angle = 90, 
    col = plasma(length(cancerTypes)), 
    main = "Proportion of 501 Cancer Lineages Represented in DepMap")
legend("left", legend=cancerTypes, fill=plasma(length(cancerTypes)))
```

Barplots of cancer mutation types

```{r, eval=TRUE, echo=FALSE}
library(gridExtra)
library(grid)

# Barplot of cancer mutation types
p1 <- ggplot(mutationCalls, aes(x=factor(Variant_Classification)))+ 
    geom_bar(stat="Count", width=0.9, fill="steelblue")+theme_minimal() +
    ggtitle("Mutation class") 

# Horizontal bar plot
p2 <- (p1 + coord_flip() + theme_gray())

# Barplot of cancer mutation types (SNP)
p3 <- ggplot(mutationCalls, aes(x=factor(Variant_Type))) +
    geom_bar(stat="Count", width=0.7, fill="steelblue")+theme_minimal() +
    ggtitle("Mutations type") 

# Horizontal bar plot
p4 <- (p3 + coord_flip() + theme_gray())

#plot as 1x2 grid
grid.arrange(p2, p4, nrow = 1, top = "Depmap Cancer Mutations")
```

